<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Sweet Dreams and Genetic Schemes: Improving CSID Variant Annotation through Integrated Computational and Genetic Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>
<script type="text/javascript">
window.PlotlyConfig = {MathJaxConfig: 'local'};
if (window.MathJax && window.MathJax.Hub && window.MathJax.Hub.Config) {window.MathJax.Hub.Config({SVG: {font: "STIX-Web"}});}
if (typeof require !== 'undefined') {
require.undef("plotly");
requirejs.config({
    paths: {
        'plotly': ['https://cdn.plot.ly/plotly-2.24.1.min']
    }
});
require(['plotly'], function(Plotly) {
    window._Plotly = Plotly;
});
}
</script>



</head>

<body class="fullcontent">

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Sweet Dreams and Genetic Schemes: Improving CSID Variant Annotation through Integrated Computational and Genetic Analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<!-- Alt title: There is limited evidence to support CSID being a rare disease. -->
<!-- Some notes on csl/biblio: csl is from https://www.zotero.org/styles so
that we can have Vancouver style, since that's what most of these peeps publish with. 'Save as link' the Vancouver file. This is all per: https://quarto.org/docs/authoring/footnotes-and-citations.html#citation-style  

Using Zotero -> Export Collection -> Better Bibtex. I've done just the .bib and also the one with files, can see what ends up being better. Suspect the .bib to be sufficient but could be fun to play around with.

NOTE: Can do @X for single citation, multiple use [@X; @Y] and will do commas etc and be lovely

Showing PDBs: https://github.com/jmbuhr/quarto-molstar
... 4 Oct 2023 not working

-->
<section id="note-lightdark-mode-button-in-top-right." class="level6">
<h6 class="anchored" data-anchor-id="note-lightdark-mode-button-in-top-right.">(Note: light/dark mode button in top right.)</h6>
<!-- `{{< mol-rcsb 3lpp >}}` -->
</section>
<section id="abstract" class="level2">
<h2 class="anchored" data-anchor-id="abstract">Abstract</h2>
<p>This proposal outlines a novel approach that leverages recent advancements in computational tools and genetic analysis to enhance variant annotation for CSID, facilitating clinical diagnosis and improving understanding of CSID prevalence. Previous studies have been been hindered by historical limitations in study design and data availability. By integrating genetic databases, protein simulation techniques, and variant prediction tools, this study aims to identify new variants of interest and contribute insights into the molecular and genetic pathology of CSID, its clinical implications and prevalence.</p>
<section id="introduction" class="level3">
<h3 class="anchored" data-anchor-id="introduction">Introduction</h3>
<p>Historical studies of CSID prevalence and its mutational landscape have been constrained by small sample sizes and homogeneous populations. Early studies that are typically cited feature sample sizes of:</p>
<ul>
<li>3 in CSID’s first description<span class="citation" data-cites="weijersDiarrhoeaCausedDeficiency1960"><sup><a href="#ref-weijersDiarrhoeaCausedDeficiency1960" role="doc-biblioref">1</a></sup></span></li>
<li>399, of which 339 were Caucasian<span class="citation" data-cites="welshIntestinalDisaccharidaseActivities1978"><sup><a href="#ref-welshIntestinalDisaccharidaseActivities1978" role="doc-biblioref">2</a></sup></span></li>
<li>56 Canadian Eskimos<span class="citation" data-cites="ellestad-sayedDisaccharideMalabsorptionDietary1978"><sup><a href="#ref-ellestad-sayedDisaccharideMalabsorptionDietary1978" role="doc-biblioref">3</a></sup></span></li>
<li>97 Greenlanders<span class="citation" data-cites="gudmand-hoyerSucraseDeficiencyGreenland1987"><sup><a href="#ref-gudmand-hoyerSucraseDeficiencyGreenland1987" role="doc-biblioref">4</a></sup></span></li>
</ul>
<p>In addition, recent findings indicate that the clinical presentation of CSID shares similarities with irritable bowel syndrome, raising concerns about accurate diagnosis<span class="citation" data-cites="henstromFunctionalVariantsSucrase2018 garcia-etxebarriaIncreasedPrevalenceRare2018a kimSucraseIsomaltaseDeficiencyPotential2020"><sup><a href="#ref-henstromFunctionalVariantsSucrase2018" role="doc-biblioref">5</a>–<a href="#ref-kimSucraseIsomaltaseDeficiencyPotential2020" role="doc-biblioref">7</a></sup></span>. Furthermore, the potential variability of CSID’s phenotype, including heterozygote carriers that experience symptoms, necessitates a more thorough investigation of its genetic and molecular causes, improvement of diagnoistic tools and a reevaluation of its prevalence<span class="citation" data-cites="huseinHeterozygotesArePotential2019 habermanCongenitalSucraseisomaltaseDeficiency2017 taskinPrevalenceCongenitalSucraseisomaltase2023"><sup><a href="#ref-huseinHeterozygotesArePotential2019" role="doc-biblioref">8</a>–<a href="#ref-taskinPrevalenceCongenitalSucraseisomaltase2023" role="doc-biblioref">10</a></sup></span>.</p>
</section>
<section id="genetic-databases-and-variant-prediction" class="level3">
<h3 class="anchored" data-anchor-id="genetic-databases-and-variant-prediction">Genetic Databases and Variant Prediction</h3>
<p>Advancements in computational biology offer an opportunity to enhance the prediciton and identification of deleterious mutations and inform diagnostics and prevalence in the population.</p>
<p>Databases such as ClinVar and gnomAD now encompass diverse and extensive population variant data<span class="citation" data-cites="chenGenomewideMutationalConstraint2022 landrumClinVarPublicArchive2016"><sup><a href="#ref-chenGenomewideMutationalConstraint2022" role="doc-biblioref">11</a>,<a href="#ref-landrumClinVarPublicArchive2016" role="doc-biblioref">12</a></sup></span>. As of 17 September 2023, gnomAD contains 3470 variants for SI, and contains 646 ClinVar variants. Of the 646 ClinVar variants, 29 are annotated as pathogenic/likely pathogenic, 66 have conflicting interpretations of pathogenicity, and 272 are of uncertain significance. Therefore, considering all gnomAD data, most entries do not have clinical annotations or have not been well characterized. Previous work analyzing gnomAD data for the Sucrase-Isomaltase (SI) gene to estimate disease prevalence only included known variants<span class="citation" data-cites="deleusseEstimatingPrevalenceCongenital2022"><sup><a href="#ref-deleusseEstimatingPrevalenceCongenital2022" role="doc-biblioref">13</a></sup></span> and did not analyze this population of uncharacterized variants.</p>
<p>GnomAD includes ClinVar clinical annotations of mutations and predictions of variant effects from the Ensembl Variant Effect Predictor (VEP)<span class="citation" data-cites="mclarenEnsemblVariantEffect2016"><sup><a href="#ref-mclarenEnsemblVariantEffect2016" role="doc-biblioref">14</a></sup></span>. Predictive tools like VEP can be employed to predict the impact of uncharacterized variants. VEP predicts type of consequence – e.g.&nbsp;missense, frameshift, intron. This is often insufficient to assess the whether a variant is deleterious.</p>
<p>Fortunately, new methods have emerged: both at the protein-ligand simulation level and some LLM-based models, one of which can be applied that predicts effects for all missense mutations<span class="citation" data-cites="brandesGenomewidePredictionDisease2023"><sup><a href="#ref-brandesGenomewidePredictionDisease2023" role="doc-biblioref">15</a></sup></span>. <strong>A preliminary analysis using this model is provided below, and predicts a number of deleterious missense mutations</strong>. This and other new models could be applied to other types of variants such as indels, stop gained or even intron variants<span class="citation" data-cites="kurosawaPDIVASPathogenicityPredictor2023 brandesGenomewidePredictionDisease2023"><sup><a href="#ref-brandesGenomewidePredictionDisease2023" role="doc-biblioref">15</a>,<a href="#ref-kurosawaPDIVASPathogenicityPredictor2023" role="doc-biblioref">16</a></sup></span>.</p>
</section>
<section id="protein-simulation-and-interaction-analysis" class="level3">
<h3 class="anchored" data-anchor-id="protein-simulation-and-interaction-analysis">Protein Simulation and Interaction Analysis</h3>
<p>An orthogonal method to large language models is simulation at the protein/molecular level. Revolutionary tools like AlphaFold<span class="citation" data-cites="jumperHighlyAccurateProtein2021"><sup><a href="#ref-jumperHighlyAccurateProtein2021" role="doc-biblioref">17</a></sup></span> and DiffDock<span class="citation" data-cites="corsoDiffDockDiffusionSteps2023"><sup><a href="#ref-corsoDiffDockDiffusionSteps2023" role="doc-biblioref">18</a></sup></span> enable the modeling and evaluation of variant-protein interactions. AlphaFold can generate mutant structures from their sequences, and then these structures and their interaction with sucrose or with other proteins can be simulated using a variety of molecular simulation technologies. A promising recent tool is DiffDock<span class="citation" data-cites="corsoDiffDockDiffusionSteps2023"><sup><a href="#ref-corsoDiffDockDiffusionSteps2023" role="doc-biblioref">18</a></sup></span>, iterations of DiffDock to enable prediction protein-protein interactions or improve prediction accuracy<span class="citation" data-cites="ketataDiffDockPPRigidProteinProtein2023 brocidiaconoPLANTAINDiffusioninspiredPose2023"><sup><a href="#ref-ketataDiffDockPPRigidProteinProtein2023" role="doc-biblioref">19</a>,<a href="#ref-brocidiaconoPLANTAINDiffusioninspiredPose2023" role="doc-biblioref">20</a></sup></span>, and more established tools such as AutoDock Vina<span class="citation" data-cites="trottAutoDockVinaImproving2010 santos-martinsAcceleratingAutoDock4GPUs2021"><sup><a href="#ref-trottAutoDockVinaImproving2010" role="doc-biblioref">21</a>,<a href="#ref-santos-martinsAcceleratingAutoDock4GPUs2021" role="doc-biblioref">22</a></sup></span>. These tools can be used to to predict the effects of SI variants on protein function, i.e.&nbsp;their interactions with sucrose, heterozygotic self-interactions, or interactions with other proteins. Integration of these tools is facilitated by the recently reported Foldy<span class="citation" data-cites="robertsFoldyWebApplication2023"><sup><a href="#ref-robertsFoldyWebApplication2023" role="doc-biblioref">23</a></sup></span>, which conveniently packages AlphaFold, DiffDock, Autodock Vina, and databases into a web app that is easily deployed.</p>
<p>However, the sensitivity of either of these tools, large language model based or protein simulation based, to distinguish benign and deleterious mutatations is not guaranteed. Cross referencing predictions from the large language models and simulation tools, and validating in lab will be necessary. Previous benchmarking of AlphaFold-generated structures with molecular docking had mixed results<span class="citation" data-cites="wongBenchmarkingAlphaFoldenabledMolecular2022"><sup><a href="#ref-wongBenchmarkingAlphaFoldenabledMolecular2022" role="doc-biblioref">24</a></sup></span>. However, emerging tools like DiffDock and PLANTAIN may provide better outcomes. DiffDock reports an accuracy, when simulating without a known binding pocket, of 29%<span class="citation" data-cites="corsoDiffDockDiffusionSteps2023"><sup><a href="#ref-corsoDiffDockDiffusionSteps2023" role="doc-biblioref">18</a></sup></span>. PLANTAIN attempts to improve this accuracy by optimzing the scoring function DiffDock uses, and accounts for binding pockets, and may be more fruitful<span class="citation" data-cites="brocidiaconoPLANTAINDiffusioninspiredPose2023"><sup><a href="#ref-brocidiaconoPLANTAINDiffusioninspiredPose2023" role="doc-biblioref">20</a></sup></span>. AutoDock tools are more established but may be less accurate<span class="citation" data-cites="corsoDiffDockDiffusionSteps2023"><sup><a href="#ref-corsoDiffDockDiffusionSteps2023" role="doc-biblioref">18</a></sup></span>; but do account for binding pockets and thus may be useful to compare predictions with<span class="citation" data-cites="trottAutoDockVinaImproving2010 santos-martinsAcceleratingAutoDock4GPUs2021"><sup><a href="#ref-trottAutoDockVinaImproving2010" role="doc-biblioref">21</a>,<a href="#ref-santos-martinsAcceleratingAutoDock4GPUs2021" role="doc-biblioref">22</a></sup></span>.</p>
</section>
<section id="genetic-analysis-and-large-language-models" class="level3">
<h3 class="anchored" data-anchor-id="genetic-analysis-and-large-language-models">Genetic Analysis and Large Language Models</h3>
<p>Novel genetic analysis tools have recently emerged, providing insights into the genetic landscape of CSID. While gnomAD and ClinVar have integrated VEP, this model is limited in the scope of its predictions, and a more recent protein language model may provide better predictions for missense variants<span class="citation" data-cites="brandesGenomewidePredictionDisease2023"><sup><a href="#ref-brandesGenomewidePredictionDisease2023" role="doc-biblioref">15</a></sup></span>. Preliminary analysis (an interesting figure is shown directly below, the full analysis is in the last section) employing this tool suggests a large number of missense variants of currently unknown consequence may be associated with CSID.</p>
<!-- This analysis can be cross-referenced with data from ClinVar, gnomAD, molecular simulations, and lab confirmation. By combining these diverse data sources, a more comprehensive picture of CSID's associated mutations and potential prevalence can be developed. -->
<p>Using this protein language model we can predict the delteriousness of missense mutations. The model outputs a score for each missense mutation, and a threshold of -7.5 to separate deleterious/benign mutations is reported<span class="citation" data-cites="brandesGenomewidePredictionDisease2023"><sup><a href="#ref-brandesGenomewidePredictionDisease2023" role="doc-biblioref">15</a></sup></span>. A lower score predicts a more severe mutation. The graph below is constructed using gnomAD data for missense variants and predictions from this model.</p>
<div class="cell" data-execution_count="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This cell is lengthy and is mostly just the preliminary analysis below all copied into one cell and outputting the relevant figure.</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_csv(<span class="st">"data/gnomAD_v2.1.1_SI_variants_17sep2023.csv"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># gnomad.head()</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># create a new column 'D' as the concatenated strings of columns 'A', 'B', and 'C'</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>df[<span class="st">"Variant_ID"</span>] <span class="op">=</span> (</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>    df[<span class="st">"Chromosome"</span>].astype(<span class="bu">str</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> <span class="st">"-"</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> df[<span class="st">"Position"</span>].astype(<span class="bu">str</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> <span class="st">"-"</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> df[<span class="st">"Reference"</span>].astype(<span class="bu">str</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> <span class="st">"-"</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> df[<span class="st">"Alternate"</span>].astype(<span class="bu">str</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co"># move the new column 'D' to the far left by reindexing the columns</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df.reindex(columns<span class="op">=</span>[<span class="st">"Variant_ID"</span>] <span class="op">+</span> df.columns.tolist())</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>include_annotations <span class="op">=</span> [<span class="st">'missense_variant'</span>]</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>mask <span class="op">=</span> df[<span class="st">'VEP Annotation'</span>].isin(include_annotations)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>df_mis <span class="op">=</span> df[mask]</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>df_esm <span class="op">=</span> pd.read_csv(<span class="st">"data/ESM1b_SI.csv"</span>) </span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> re</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> Bio.SeqUtils <span class="im">import</span> seq1, seq3</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="kw">class</span> seqConversion:</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>  <span class="kw">def</span> <span class="fu">__init__</span>(<span class="va">self</span>) <span class="op">-&gt;</span> <span class="va">None</span>:</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>      <span class="cf">pass</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># intended to be private method, intermediate for convertGnomadHGVS below.      </span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>  <span class="kw">def</span> _parseGnomadHGVS(<span class="va">self</span>, hgvsEntry):</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Define a regex pattern to match the format "stringNUMBERstring"</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>    pattern <span class="op">=</span> <span class="vs">r'([A-Za-z]+)(\d+)([A-Za-z]+)'</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use re.match to find the first match in the input string, need to account for multiple and check the duplicates if exist</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>    match <span class="op">=</span> re.match(pattern, hgvsEntry)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> match: <span class="co"># which should be, but good in case weird data i don't anticipate</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Extract the three parts from the match object</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>        aa_wt <span class="op">=</span> match.group(<span class="dv">1</span>)</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>        pos <span class="op">=</span> match.group(<span class="dv">2</span>)</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>        aa_mut <span class="op">=</span> match.group(<span class="dv">3</span>)</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Convert the numbers to an integer (if needed)</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>        pos <span class="op">=</span> <span class="bu">int</span>(pos)</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> aa_wt, pos, aa_mut</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span><span class="al">FIXME</span><span class="co">: May need more handling cases, so far this is working on SI data 22 Sep 2023</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Handle the case where no match is found</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> <span class="va">None</span>, <span class="va">None</span>, <span class="va">None</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>  <span class="kw">def</span> convertGnomadHGVS(<span class="va">self</span>, hgvsEntry):</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span><span class="al">NOTE</span><span class="co">: This assumes we're only passing in the protein-level variants, which, should be the only case this function is called, i.e. missense variants</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>    <span class="co"># strip 'p.'</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>    hgvsEntry <span class="op">=</span> hgvsEntry.lstrip(<span class="st">'p.'</span>)</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get the gnomad formatted variables</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>    aa_wt, pos, aa_mut <span class="op">=</span> <span class="va">self</span>._parseGnomadHGVS(hgvsEntry)</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> aa_wt <span class="kw">is</span> <span class="va">None</span> <span class="kw">or</span> pos <span class="kw">is</span> <span class="va">None</span> <span class="kw">or</span> aa_mut <span class="kw">is</span> <span class="va">None</span>:</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>            <span class="cf">return</span> <span class="va">None</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>      <span class="co"># codes to one-letter codes; from BioSeq import above</span></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>      aa_wt1 <span class="op">=</span> seq1(aa_wt)</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>      aa_mut1 <span class="op">=</span> seq1(aa_mut)</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>      <span class="co"># assemble the esm1b format of gnomad entry</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>      newFormat <span class="op">=</span> <span class="ss">f"</span><span class="sc">{</span>aa_wt1<span class="sc">}{</span>pos<span class="sc">}{</span>aa_mut1<span class="sc">}</span><span class="ss">"</span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>      <span class="cf">return</span> newFormat</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>  <span class="kw">def</span> getProteinVariants(hgvsEntry):</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>    p_pattern <span class="op">=</span> <span class="vs">r"p.*"</span>  <span class="co"># Use .* to match zero or more occurrences of 'p'</span></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>    matches <span class="op">=</span> re.findall(p_pattern, hgvsEntry)</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> matches:</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> matches</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Handle the case where no match is found</span></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> <span class="va">None</span></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a><span class="co"># Converter = seqConversion()</span></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="co"># df_mis['HGVS_ESM1b']  = df_mis['HGVS Consequence'].apply(Converter.convertGnomadHGVS)</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="co"># to deal with error:</span></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>Converter <span class="op">=</span> seqConversion()</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a><span class="co"># df_mis.loc[:, 'HGVS_ESM1b'] = df_mis['HGVS Consequence'].apply(Converter.convertGnomadHGVS)</span></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>df_mis <span class="op">=</span> df_mis.assign(HGVS_ESM1b<span class="op">=</span>df_mis[<span class="st">'HGVS Consequence'</span>].<span class="bu">apply</span>(Converter.convertGnomadHGVS))</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>merged_df <span class="op">=</span> df_mis.merge(df_esm[[<span class="st">'variant'</span>, <span class="st">'score'</span>]], left_on<span class="op">=</span><span class="st">'HGVS_ESM1b'</span>, right_on<span class="op">=</span><span class="st">'variant'</span>, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> plotly.express <span class="im">as</span> px</span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the 'score' column from the merged DataFrame</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>scores <span class="op">=</span> merged_df[<span class="st">"score"</span>]</span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a>all_known_path <span class="op">=</span> df[<span class="st">'ClinVar Clinical Significance'</span>].isin([<span class="st">'Pathogenic'</span>, <span class="st">'Likely pathogenic'</span>, <span class="st">'Pathogenic/Likely pathogenic  '</span>])</span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>akp_df <span class="op">=</span> df[all_known_path]</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>pathogenic_missense_variants <span class="op">=</span> merged_df[<span class="st">'ClinVar Clinical Significance'</span>].isin([<span class="st">'Pathogenic'</span>, <span class="st">'Likely pathogenic'</span>,]) <span class="co"># # adds 23 more samples that have varying scores</span></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>pmv_df <span class="op">=</span> merged_df[pathogenic_missense_variants]</span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>pmv_df <span class="op">=</span> pmv_df.sort_values(by<span class="op">=</span><span class="st">'score'</span>, ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter, get benign bariants</span></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a>benign_missense_variants <span class="op">=</span> merged_df[<span class="st">'ClinVar Clinical Significance'</span>].isin([<span class="st">'Benign'</span>, <span class="st">'Benign/Likely benign'</span>, <span class="st">'Likely benign'</span>])</span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a><span class="co"># bmv: benign missense variants</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a>bmv_df <span class="op">=</span> merged_df[benign_missense_variants]</span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>bmv_df <span class="op">=</span> bmv_df.sort_values(by<span class="op">=</span><span class="st">'score'</span>, ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the threshold</span></span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a>threshold <span class="op">=</span> <span class="op">-</span><span class="fl">7.5</span></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new feature 'below threshold' based on the 'score' column</span></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a>merged_df[<span class="st">'below threshold'</span>] <span class="op">=</span> merged_df[<span class="st">'score'</span>].<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">"yes"</span> <span class="cf">if</span> x <span class="op">&lt;</span> threshold <span class="cf">else</span> <span class="st">"no"</span>)</span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> plotly.express <span class="im">as</span> px</span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> px.histogram(</span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>    merged_df,</span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a>    x<span class="op">=</span><span class="st">"score"</span>,</span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a>    color <span class="op">=</span> <span class="st">"below threshold"</span>,</span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a>    nbins<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a>    opacity<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a>    labels<span class="op">=</span>{<span class="st">"score"</span>: <span class="st">"Scores"</span>},</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a>    title<span class="op">=</span><span class="st">"Distribution of Scores Within Gnomad"</span>,</span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a><span class="co"># Add markers or lines for known pathogenic variants</span></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx, row <span class="kw">in</span> pmv_df.iterrows():</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a>    fig.add_shape(<span class="bu">type</span><span class="op">=</span><span class="st">'line'</span>, x0<span class="op">=</span>row[<span class="st">'score'</span>], x1<span class="op">=</span>row[<span class="st">'score'</span>], y0<span class="op">=</span><span class="dv">0</span>, y1<span class="op">=</span><span class="dv">1</span>, opacity <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a>                  xref<span class="op">=</span><span class="st">'x'</span>, yref<span class="op">=</span><span class="st">'paper'</span>, line<span class="op">=</span><span class="bu">dict</span>(color<span class="op">=</span><span class="st">'red'</span>))</span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a><span class="co"># Add markers/lines for benign variants</span></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx, row <span class="kw">in</span> bmv_df.iterrows():</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a>    fig.add_shape(<span class="bu">type</span><span class="op">=</span><span class="st">'line'</span>, x0<span class="op">=</span>row[<span class="st">'score'</span>], x1<span class="op">=</span>row[<span class="st">'score'</span>], y0<span class="op">=</span><span class="dv">0</span>, y1<span class="op">=</span><span class="dv">1</span>, opacity <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>                  xref<span class="op">=</span><span class="st">'x'</span>, yref<span class="op">=</span><span class="st">'paper'</span>, line<span class="op">=</span><span class="bu">dict</span>(color<span class="op">=</span><span class="st">'green'</span>))</span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a><span class="co"># Customize the layout</span></span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>fig.update_layout(bargap<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a><span class="co"># Show the plot</span></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a>fig.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">

<div>                            <div id="0a3b8230-77b4-4d7e-80c4-5e15a95ecafc" class="plotly-graph-div" style="height:525px; width:100%;"></div>            <script type="text/javascript">                require(["plotly"], function(Plotly) {                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById("0a3b8230-77b4-4d7e-80c4-5e15a95ecafc")) {                    Plotly.newPlot(                        "0a3b8230-77b4-4d7e-80c4-5e15a95ecafc",                        [{"alignmentgroup":"True","bingroup":"x","hovertemplate":"below threshold=no\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"no","marker":{"color":"#636efa","opacity":1,"pattern":{"shape":""}},"name":"no","nbinsx":20,"offsetgroup":"no","orientation":"v","showlegend":true,"x":[-2.357,-3.158,-4.747,-6.465,-4.411,-2.642,-2.574,-4.16,-2.664,-5.22,-4.315,-2.647,-4.838,-5.463,-3.605,-5.089,-1.802,-6.435,-5.614,-7.055,-3.803,-3.651,-2.097,-2.097,-2.583,-2.917,-4.817,-4.282,-1.665,-4.14,-5.018,-5.784,-1.888,-3.553,-4.4,-4.379,-5.445,-5.233,-1.861,-3.652,-4.29,-2.927,-5.93,-3.845,-5.32,-6.68,-7.299,-5.275,-4.09,-2.355,-5.678,-5.38,-7.243,-6.873,-3.736,-3.696,-0.588,-6.241,-4.025,-4.552,-7.067,-5.752,-5.66,-6.18,-4.738,-4.634,-6.893,-5.901,-3.792,-2.504,-4.959,-4.465,-6.256,-4.194,-2.136,-3.267,-7.034,-5.563,-6.154,-5.586,-6.942,-6.252,-1.22,-0.875,-6.669,-6.83,-7.4,-4.799,-6.071,-5.819,-4.875,-4.966,-3.801,-1.363,-4.784,-4.141,-0.146,-5.93,-5.355,-3.337,-4.273,-4.295,-6.691,-7.104,-5.757,-6.103,-3.456,-3.96,-5.509,-5.597,-6.458,-4.019,-1.651,-4.164,-4.958,-6.279,-1.174,-5.266,-3.31,-2.449,-3.437,-3.37,-5.424,-4.533,-3.726,-1.678,-3.703,-4.311,-4.311,-4.204,-4.875,-4.562,-4.765,-5.602,-6.627,-5.602,-5.359,-6.283,-5.701,-6.106,-4.523,-6.974,-4.179,-6.777,-3.717,-6.039,-4.971,-2.089,-5.754,-3.693,-6.249,-6.26,-7.117,-4.421,-5.507,-7.016,-6.695,-4.496,-5.076,-7.046,-5.269,-6.813,-5.199,-3.485,-6.675,-2.514,-6.811,-5.671,-3.512,-5.09,-7.14,-6.537,-6.629,-2.284,-7.034,-5.314,-5.665,-2.842,-4.654,-5.243,-1.464,-6.988,-6.166,-6.357,-2.104,-6.272,-3.38,-5.611,-1.82,-4.786,-5.322,-6.744,0.526,-7.498,-5.663,-6.911,-5.276,-0.717,-6.373,-6.421,-6.996,-7.492,-7.161,-5.755,-6.713,-4.137,-6.684,-0.094,-5.873,-3.356,-6.721,-4.742,-7.346,-6.596,-2.28,-4.499,-2.962,-3.724,-6.978,-5.262,-5.534,-7.145,-6.563,-4.352,-4.354,-5.935,-2.218,-5.439,-4.415,-2.322,-5.788,-5.223,-5.672,-7.346,-4.779,-6.88,-6.611,-5.39,-7.396,-6.468,-3.656,-6.418,-6.574,-2.965,-3.015,-5.71,-4.362,-5.081,-5.267,-7.018,-4.196,-3.973,-1.552,-3.563,-7.17,-1.729,-2.457,-3.222,-2.452,-7.154,-4.267,-3.873,-6.937,-6.992,-2.751,-3.347,-5.54,-1.597,-7.439,-2.21,-4.767,-1.746,-0.606,-1.801,-3.403,-4.553,-7.251,-2.248,-3.976,-4.442,-6.106,-1.093,-0.167,-3.724,-6.691,-3.473,-7.054,-6.322,-4.187,-3.267,-5.382,-2.495,-2.241,-5.049,-2.986,-3.132,-5.834,-5.844,-4.816,-6.279,-3.973,-2.651,-6.652,-5.146,-6.35,-4.545,-4.539,-3.794,-3.994,-6.792,-5.987,-4.19,0.054,-4.581,-3.823,-5.998,-4.931,-3.984,-4.207,-2.653,-3.831,-4.251,-4.187,-4.071,-5.279,-6.081,-3.772,-7.484,-5.401,-5.58,-3.865,-2.766,-5.362,-3.8,-3.162,-3.643,-6.425,-3.848,-3.026,-6.827,-2.627,-7.294,-5.5,-7.478,-7.362,-2.005,-5.417,-6.051,-4.526,-1.816,-7.22,-6.877,-4.739,-5.897,-5.145,-4.525,-6.409,-5.987,-5.765,-6.92,-5.404,-3.839,-3.858,-6.014,-7.164,-4.708,-7.373,-5.937,-6.735,-6.909,-5.905,-4.278,-6.676,-6.482,-6.562,-5.747,-1.396,-3.534,-6.925,-3.646,-4.292,-5.696,-6.021,-3.829,-5.116,-5.996,-5.368,-7.375,-5.905,-7.147,-6.475,-7.083,-5.955,-5.366,-5.982,-7.109,-3.924,-3.291,-3.795,-5.17,-5.841,-6.923,-7.003,-7.12,-4.069,-7.229,-6.608,-6.3,-5.75,-6.932,-3.123,-6.871,-3.221,-7.372,-3.332,-4.215,-4.068,-5.003,-6.621,-5.759,-3.156,-4.175,-5.725,-4.868,-6.811,-6.896,-4.417,-2.901,-7.467,-1.607,-5.213,-4.804,-2.793,-6.332,-4.294,-2.269,-5.054,-6.81,-6.035,-4.465,-6.113,-6.537,-3.416,-2.261,-5.855,-5.193,-1.649,-5.12,-2.342,-4.617,-5.331,-6.92,-2.36,-3.589,-7.287,-3.878,-4.711,-5.117,-4.505,-5.835,-7.19,-4.085,-5.859,-5.576,-6.848,-4.881,-3.961,-2.24,-5.296,-6.853,-6.002,-6.846,-6.879,-6.404,-6.851,-7.163,-6.631,-3.082,-7.452,-1.704,-7.0,-2.196,-4.842,-5.352,-2.772,-6.514,-5.483,-4.052,-0.656,-5.49,-4.172,-4.032,-4.402,-2.645,-3.587,-3.967,-6.08,-3.788,-7.087,-7.091,-3.307,-3.905,-4.278,-5.457,-6.417,-7.354,-5.021,-7.147,-3.309,-6.034,-6.501,-4.932,-3.774,-4.067,-4.735,-5.743,-1.885,-5.563,-3.428,-6.225,-4.056,-7.377,-3.661,-4.368,-3.474,-5.738,-5.712,-4.933,-3.029,-3.438,-6.656,-5.117,-2.039,-5.541,-4.755,-6.711,-7.294,-4.084,-6.396,-0.409,-4.286,-2.781,-5.03,-5.652,-6.609,-6.977,-7.094,-3.833,-0.032,-6.704,-6.847,-1.923,-5.294,-7.273,-6.088,-4.102,-3.975,-4.164,-3.318,-4.576,-2.676,-6.022,-5.601,-5.643,-5.525,-3.023,-6.079,-4.865,-3.99,-4.722,-5.823,-3.726,-5.238,-3.963,-2.898,-3.349,-5.426,-5.61,-5.778,-4.291,-7.498,-0.102,-4.85,-6.436,-2.619,-3.935,-5.406,-4.897,-3.092,-2.964,-5.346,-3.307,-5.7,-3.231,-6.14,-6.713,-5.691,-7.045,-5.403,-4.323,-5.201,-6.813,-3.955,-2.163,-2.724,-4.913,-6.316,-7.303,-6.888,-6.332],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"below threshold=yes\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"yes","marker":{"color":"#EF553B","opacity":1,"pattern":{"shape":""}},"name":"yes","nbinsx":20,"offsetgroup":"yes","orientation":"v","showlegend":true,"x":[-9.24,-8.665,-7.574,-7.917,-8.163,-13.305,-10.512,-7.916,-8.534,-7.886,-9.789,-15.265,-10.476,-11.891,-8.499,-14.373,-8.062,-8.676,-9.04,-9.708,-10.182,-9.738,-8.31,-11.722,-15.247,-13.009,-10.524,-11.179,-9.417,-9.76,-9.624,-8.668,-8.922,-10.988,-7.562,-11.494,-9.311,-11.546,-7.82,-10.55,-9.838,-9.247,-10.043,-8.456,-12.181,-12.991,-12.021,-9.904,-11.098,-13.547,-9.939,-10.267,-12.241,-14.01,-8.335,-9.771,-8.273,-9.418,-11.672,-9.5,-8.853,-10.275,-9.819,-9.548,-11.077,-11.328,-9.742,-13.477,-10.657,-8.58,-8.015,-13.313,-11.903,-11.24,-12.293,-11.623,-12.76,-13.53,-13.247,-13.102,-12.073,-14.094,-8.465,-11.546,-11.198,-8.086,-11.091,-9.862,-13.533,-8.62,-11.427,-7.574,-11.599,-10.674,-10.946,-13.568,-8.637,-8.05,-11.013,-14.583,-12.225,-14.027,-12.347,-10.452,-7.88,-12.914,-8.472,-8.889,-11.494,-10.518,-9.659,-13.291,-11.674,-9.316,-8.399,-8.697,-8.541,-9.049,-13.129,-9.681,-11.629,-10.342,-11.822,-10.11,-12.838,-13.534,-7.662,-10.334,-13.632,-12.706,-12.976,-7.879,-11.443,-8.365,-9.676,-12.839,-11.058,-9.07,-8.027,-9.008,-8.0,-13.131,-9.68,-15.569,-11.647,-8.785,-8.965,-7.556,-11.751,-8.08,-9.641,-12.069,-9.275,-10.949,-10.87,-11.428,-12.238,-12.902,-10.938,-11.083,-11.256,-7.578,-12.091,-8.489,-9.929,-8.817,-15.482,-11.748,-11.981,-10.585,-10.848,-11.517,-11.661,-9.824,-11.947,-10.392,-9.367,-11.271,-8.422,-11.223,-8.817,-8.737,-10.995,-11.932,-10.094,-11.256,-11.52,-9.854,-10.727,-13.123,-11.193,-10.97,-13.024,-12.188,-14.2,-8.416,-8.411,-7.704,-11.374,-12.9,-13.95,-12.562,-9.453,-8.363,-7.852,-10.377,-7.991,-10.563,-10.447,-8.282,-9.9,-7.508,-7.762,-8.468,-11.303,-9.586,-12.56,-9.487,-13.105,-10.482,-11.55,-9.091,-8.725,-9.726,-12.158,-9.024,-10.663,-10.15,-10.273,-9.522,-8.242,-7.635,-9.373,-9.153,-7.676,-13.393,-9.851,-12.62,-12.029,-8.022,-12.013,-8.943,-7.658,-11.552,-14.427,-9.083,-11.157,-8.77,-10.136,-8.306,-8.273,-7.741,-8.565,-11.868,-8.426,-7.655,-9.3,-8.171,-11.548,-8.497,-7.752,-10.428,-11.863,-7.893,-8.689,-8.848,-7.854,-7.726,-9.012,-8.291,-8.101,-13.475,-11.549,-8.22,-8.335,-9.116,-8.627,-11.381,-12.17,-12.786,-8.47,-10.559,-11.542,-12.716,-11.963,-14.742,-10.615,-8.785,-11.04,-7.967,-11.117,-14.417,-13.017,-9.716,-8.779,-8.263,-8.536,-9.33,-10.854,-8.052,-9.6,-7.557,-8.242,-9.889,-11.968,-9.061,-12.852,-12.468,-7.641,-8.399,-10.492,-9.247,-8.812,-13.295,-9.022,-9.792,-12.924,-11.328,-10.777,-9.018,-12.728,-12.849,-8.444,-9.314,-11.753,-10.268,-10.353,-11.109,-11.36,-9.486,-9.201,-9.036,-10.871,-11.589,-15.203,-12.431,-11.373,-10.986,-9.601,-9.692,-10.412,-10.756,-13.468,-8.088,-16.026,-11.124,-12.054,-10.46,-13.77,-10.232,-10.313,-9.186,-9.476,-11.638,-12.864,-11.669,-12.337,-7.901,-10.341,-13.021,-9.337,-12.035,-8.054,-9.233,-9.602,-14.151,-8.801,-14.893,-10.118,-11.75,-12.047,-11.265,-8.357,-11.6,-9.324,-12.087,-9.151,-9.162,-11.664,-11.478,-7.558,-9.954,-8.792,-11.965,-7.806,-11.3,-10.734,-10.615,-11.081,-8.804,-11.892,-10.28,-12.073,-7.619,-12.423,-13.225,-11.773,-9.801,-13.359,-12.057,-12.298,-7.669,-10.797,-9.935,-7.86,-8.258,-10.632,-7.874,-10.559,-10.22,-13.695,-14.58,-7.88,-12.974,-9.019,-7.772,-8.035,-11.006,-13.892,-8.158,-10.313,-7.88,-10.531,-13.687,-11.658,-10.563,-11.504,-8.949,-11.694,-11.627,-9.643,-10.4,-9.002,-8.994,-8.427,-11.391,-13.092,-11.536,-8.956,-8.007,-11.63,-10.165,-7.669,-8.013,-8.974,-10.559,-9.658,-9.173,-8.037,-10.074,-11.076,-7.602,-13.272,-12.432,-10.839,-10.279,-12.282,-9.286,-7.954,-13.656,-10.33,-12.237,-10.782,-14.762,-9.651,-8.489,-9.816,-9.398,-7.686,-11.034,-9.913,-12.377,-7.766,-7.89,-9.394,-11.12,-8.22,-11.2,-10.235,-9.426,-8.413,-11.158,-11.265,-10.881,-12.403,-9.493,-8.772,-10.519,-9.713,-8.218,-9.031,-14.119,-12.776,-7.759,-12.196,-11.898,-13.732,-11.867,-9.465,-8.643,-13.06,-10.139,-7.957,-7.779,-7.833,-11.409,-10.97,-8.151,-7.811,-9.097,-8.876,-11.577,-10.876,-7.663,-10.808,-10.188,-9.538,-11.336,-11.228,-8.045,-10.217,-7.857,-9.734,-9.506,-9.528,-10.196,-12.899,-10.206,-8.397,-16.797,-8.858],"xaxis":"x","yaxis":"y","type":"histogram"}],                        {"template":{"data":{"histogram2dcontour":[{"type":"histogram2dcontour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"choropleth":[{"type":"choropleth","colorbar":{"outlinewidth":0,"ticks":""}}],"histogram2d":[{"type":"histogram2d","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmap":[{"type":"heatmap","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmapgl":[{"type":"heatmapgl","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"contourcarpet":[{"type":"contourcarpet","colorbar":{"outlinewidth":0,"ticks":""}}],"contour":[{"type":"contour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"surface":[{"type":"surface","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"mesh3d":[{"type":"mesh3d","colorbar":{"outlinewidth":0,"ticks":""}}],"scatter":[{"fillpattern":{"fillmode":"overlay","size":10,"solidity":0.2},"type":"scatter"}],"parcoords":[{"type":"parcoords","line":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolargl":[{"type":"scatterpolargl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"bar":[{"error_x":{"color":"#2a3f5f"},"error_y":{"color":"#2a3f5f"},"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"bar"}],"scattergeo":[{"type":"scattergeo","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolar":[{"type":"scatterpolar","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"histogram":[{"marker":{"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"histogram"}],"scattergl":[{"type":"scattergl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatter3d":[{"type":"scatter3d","line":{"colorbar":{"outlinewidth":0,"ticks":""}},"marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattermapbox":[{"type":"scattermapbox","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterternary":[{"type":"scatterternary","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattercarpet":[{"type":"scattercarpet","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"carpet":[{"aaxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"baxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"type":"carpet"}],"table":[{"cells":{"fill":{"color":"#EBF0F8"},"line":{"color":"white"}},"header":{"fill":{"color":"#C8D4E3"},"line":{"color":"white"}},"type":"table"}],"barpolar":[{"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"barpolar"}],"pie":[{"automargin":true,"type":"pie"}]},"layout":{"autotypenumbers":"strict","colorway":["#636efa","#EF553B","#00cc96","#ab63fa","#FFA15A","#19d3f3","#FF6692","#B6E880","#FF97FF","#FECB52"],"font":{"color":"#2a3f5f"},"hovermode":"closest","hoverlabel":{"align":"left"},"paper_bgcolor":"white","plot_bgcolor":"#E5ECF6","polar":{"bgcolor":"#E5ECF6","angularaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"radialaxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"ternary":{"bgcolor":"#E5ECF6","aaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"baxis":{"gridcolor":"white","linecolor":"white","ticks":""},"caxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"coloraxis":{"colorbar":{"outlinewidth":0,"ticks":""}},"colorscale":{"sequential":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"sequentialminus":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"diverging":[[0,"#8e0152"],[0.1,"#c51b7d"],[0.2,"#de77ae"],[0.3,"#f1b6da"],[0.4,"#fde0ef"],[0.5,"#f7f7f7"],[0.6,"#e6f5d0"],[0.7,"#b8e186"],[0.8,"#7fbc41"],[0.9,"#4d9221"],[1,"#276419"]]},"xaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"yaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"scene":{"xaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"yaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"zaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2}},"shapedefaults":{"line":{"color":"#2a3f5f"}},"annotationdefaults":{"arrowcolor":"#2a3f5f","arrowhead":0,"arrowwidth":1},"geo":{"bgcolor":"white","landcolor":"#E5ECF6","subunitcolor":"white","showland":true,"showlakes":true,"lakecolor":"white"},"title":{"x":0.05},"mapbox":{"style":"light"}}},"xaxis":{"anchor":"y","domain":[0.0,1.0],"title":{"text":"Scores"}},"yaxis":{"anchor":"x","domain":[0.0,1.0],"title":{"text":"count"}},"legend":{"title":{"text":"below threshold"},"tracegroupgap":0},"title":{"text":"Distribution of Scores Within Gnomad"},"barmode":"relative","shapes":[{"line":{"color":"red"},"opacity":0.4,"type":"line","x0":-11.661,"x1":-11.661,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"red"},"opacity":0.4,"type":"line","x0":-10.267,"x1":-10.267,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"red"},"opacity":0.4,"type":"line","x0":-8.792,"x1":-8.792,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-11.627,"x1":-11.627,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-10.531,"x1":-10.531,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-9.097,"x1":-9.097,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-8.77,"x1":-8.77,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-8.397,"x1":-8.397,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-7.857,"x1":-7.857,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-7.578,"x1":-7.578,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-7.14,"x1":-7.14,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.877,"x1":-6.877,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.744,"x1":-6.744,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.691,"x1":-6.691,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.417,"x1":-6.417,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.373,"x1":-6.373,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.256,"x1":-6.256,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-5.5,"x1":-5.5,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-5.457,"x1":-5.457,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-5.382,"x1":-5.382,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-5.089,"x1":-5.089,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-4.799,"x1":-4.799,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-4.738,"x1":-4.738,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-4.634,"x1":-4.634,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-3.96,"x1":-3.96,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-3.652,"x1":-3.652,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-2.664,"x1":-2.664,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-2.647,"x1":-2.647,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-2.449,"x1":-2.449,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-1.885,"x1":-1.885,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-1.597,"x1":-1.597,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-1.22,"x1":-1.22,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-0.146,"x1":-0.146,"xref":"x","y0":0,"y1":1,"yref":"paper"}],"bargap":0.1},                        {"responsive": true}                    ).then(function(){
                            
var gd = document.getElementById('0a3b8230-77b4-4d7e-80c4-5e15a95ecafc');
var x = new MutationObserver(function (mutations, observer) {{
        var display = window.getComputedStyle(gd).display;
        if (!display || display === 'none') {{
            console.log([gd, 'removed!']);
            Plotly.purge(gd);
            observer.disconnect();
        }}
}});

// Listen for the removal of the full notebook cells
var notebookContainer = gd.closest('#notebook-container');
if (notebookContainer) {{
    x.observe(notebookContainer, {childList: true});
}}

// Listen for the clearing of the current output cell
var outputEl = gd.closest('.output');
if (outputEl) {{
    x.observe(outputEl, {childList: true});
}}

                        })                };                });            </script>        </div>
</div>
</div>
<p>In this preliminary analysis, ClinVar annotated variants that are benign/likely benign and pathogenic/likely pathogenic are shown with green/red lines overlaid on the distribution of variants in gnomAD and their predicted deleteriousness. There are limited missense annotated variants compared to the overall dataset – but a trend does appear amongst benign mutations to be above the score threshold and be correctly separated.</p>
<p>Overall, there are 531 missense mutations that fall below the threshold and are predicted to be deleterious. To validate these results, a good approach may be to start from the most deleterious (negatively scored) predicted mutation and assess variants with increasingly positive scores.</p>
</section>
<section id="background-and-collaboration" class="level3">
<h3 class="anchored" data-anchor-id="background-and-collaboration">Background and Collaboration</h3>
<p>My background includes an MS in bioinformatics where I gained experience in protein analysis, 2 years in pharameceutical R&amp;D at Eli Lilly performing physicochemical testing on antibodies and peptides to assess stability, and a BS in Chemical Engineering, BS in Molecular and Cellular Biology and BA in Biochemistry.</p>
<p>My motivation to study CSID stems from my patent experience, where CSID emerged as a potential explanation. I identified a gap in the literature and saw an opportunity to leverage new computational tools to improve patient outcomes. I’m open to share more details about my journey. The patient experience with the disease as a candidate has been arduous and contributing to better care for others would be deeply gratifying.</p>
<p>The preparation of this proposal has also been highly informative for career aspirations. I’m actively considering pursuing doctoral studies (MD, PhD, or an MD/PhD) and this project aligns well with my goal of connecting research and healthcare.</p>
<p>The proposed project is intricate and collaboration is indispensible. Seeking guidance and partnerships with experts has been a priority from the start. Several opportunities for collaboration include:</p>
<ul>
<li>Collaborating with Dr.&nbsp;D’Amato, and Dr.&nbsp;Naim for lab confirmation and disease expertise. I’ve also reached out to Dr.&nbsp;Treem, who authored some earlier reviews. Together lab confirmation can be facilitated/organized and a thorough understanding and robust DOE can be achieved.</li>
<li>Engage with the Foldy development team (Keasling Lab/Berkeley National Lab) for features such as DiffDock-PP or PLANTAIN that may be used and integrated into the software; the development in this project may benefit others, or integration of these tools can be conducted in parallel as part of other efforts.</li>
<li>Seek input from the authors of DiffDock and PLANTAIN, and other molecular simulation experts. Their insights would be invaluable for optimizing the approach and understanding the strengths and limitations of these tools.</li>
</ul>
</section>
</section>
<section id="costbudget" class="level2">
<h2 class="anchored" data-anchor-id="costbudget">Cost/Budget</h2>
<p><em>This section to be finalized after nailing down lab confirmation</em></p>
<p><strong>(Ask about lab confirmation costs)</strong></p>
<p><strong>(Ask about salary/stipend/consulting? idk how this works with QOL)</strong></p>
<p>The cost estimates stated here are retrieved from Google Cloud Pricing Calculator on 26 Sep 2023.</p>
<p>Using the Foldy architecture/out of the box solution for AlphaFold and molecular simulation, computation will run on Google Cloud. The SI complex is about 1500 residues; Foldy documentation suggests that 3000 residue complexes be run on a machine type that costs $3700 a month. Therefore a mid-range machine may be appropriate, one of which costs $2682 a month. Low-range machines of about $420-1509 a month are also available but are unlikely to have sufficient resources to run the SI complex. The cost estimates are attached as pdfs.</p>
<p>The scope of the final project will dictate how many variants are run and the implementation of other tools beyond what is included in Foldy (Foldy includes AlphaFold, DiffDock and AutoDock Vina along with database integration). These tools will likely run on the same machine but require additional development time.</p>
</section>
<section id="timeline-and-milestones" class="level2">
<h2 class="anchored" data-anchor-id="timeline-and-milestones">Timeline and Milestones</h2>
<p><em>This section to be finalized after nailing down lab confirmation</em></p>
<p>(<strong>Ask about lab confirmation timeline</strong>)</p>
<p>This study proposes computational methods and laboratory confirmation. The LLMs are relatively cheap to run, whereas the infrastructure and machines for AlphaFold and molecular simulations are more intensive.</p>
<p>Further, there are models not included in Foldy. Timeline to implementation of different methods will be influenced by potential collaboration partners and documentation of methods, namely PLANTAIN and other LLM. Using Foldy as-is however, deployment is finished in a few days (time for machine to boot and download relevant databases), after which simulations can be run. Initially, scripts would be developed to construct sequences from Ensembl/UniProt consensus sequence, gnomAD data, or other data sources.</p>
<p>Laboratory confirmation can occur after all results have been generated, or after smaller bolus runs of the simulations. This strategy may be superior to inform model optimization and feasibility.</p>
<p>A reasonable order of events or timeline may look like the below:</p>
<ol type="1">
<li>Validate LLM model: <strong>Lab validation, timeline per variant? of LLM missense model(??)</strong></li>
<li>Initial protein simulation analysis: setup of as-is of Foldy and development, validation of scripts to generate mutant protein sequences (2 weeks)</li>
<li>Running a small sample, validate protein simulation strategy; unknown how much time to generate each SI complex, molecular simulation should be relatively quick</li>
<li><strong>Lab confirmation/validation of Fold/docking results(??)</strong></li>
<li>Integration and validation of other models, e.g.&nbsp;other LLM, or other simulation like PLANTAIN, DiffDock-PP</li>
<li><strong>More lab confirmation</strong></li>
</ol>
<p>Therefore within <strong>dev/simulation + lab confirmation</strong> some idea of validity of models and feasibility can be achieved. Then, conservatively, within <strong>… a few months of development and further lab confirmation…?</strong> the overall feasibilty and scope of the project can be assessed.</p>
<!-- **Conclusion:**
By synergistically combining computational analysis, protein simulation, and genetic assessment, this research proposes to provide a more accurate estimate of the prevalence of CSID. The cumulative results from these interdisciplinary approaches will offer a stronger foundation for understanding the genetic and molecular basis of CSID and its broader implications for clinical diagnosis and management.

 -->
</section>
<section id="preliminary-analysis" class="level1">
<h1>Preliminary Analysis</h1>
<p>As of 17 September 2023, the gnomAD v2.1.1 dataset contains 3470 variants, and ClinVar contains 988 variants. The ClinVar annotations are:</p>
<div class="cell" data-execution_count="2">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">"""</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co">This cell does some analysis on gnomAD data for the SI gene. </span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">As of 17 Sep 2023, the gnomAD page for SI, https://gnomad.broadinstitute.org/gene/ENSG00000090402?dataset=gnomad_r2_1</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co">contains 988 ClinVar variants, and 3470 gnomAD variants. Let's look at how many have been clinically annotated, and how many are present in gnomAD. Then we'll cross-reference with the ESMB1 data below to suggest how many could be deleterious, via that model.</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">"""</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_csv(<span class="st">"data/gnomAD_v2.1.1_SI_variants_17sep2023.csv"</span>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co"># gnomad.head()</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="co"># create a new column 'D' as the concatenated strings of columns 'A', 'B', and 'C'</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>df[<span class="st">"Variant_ID"</span>] <span class="op">=</span> (</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    df[<span class="st">"Chromosome"</span>].astype(<span class="bu">str</span>)</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> <span class="st">"-"</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> df[<span class="st">"Position"</span>].astype(<span class="bu">str</span>)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> <span class="st">"-"</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> df[<span class="st">"Reference"</span>].astype(<span class="bu">str</span>)</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> <span class="st">"-"</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span> df[<span class="st">"Alternate"</span>].astype(<span class="bu">str</span>)</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="co"># move the new column 'D' to the far left by reindexing the columns</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df.reindex(columns<span class="op">=</span>[<span class="st">"Variant_ID"</span>] <span class="op">+</span> df.columns.tolist())</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="co"># display the updated dataframe</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a><span class="co"># df.head()</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a><span class="co"># demonstrates 3470 variants in gnomAD as of this date. </span></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>df.describe()</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a><span class="co"># 640 variants for this disease, of which there are the 8 unique annotations</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a><span class="co"># df["ClinVar Clinical Significance"].describe()</span></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a><span class="co"># demonstrates the majority are of unknown consequence and could be examined </span></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"The below is the makeup of ClinVar variants...</span><span class="ch">\n</span><span class="st">"</span>,</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>df[<span class="st">"ClinVar Clinical Significance"</span>].value_counts()) <span class="co"># /640, 17Sep2023</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>The below is the makeup of ClinVar variants...
 ClinVar Clinical Significance
Uncertain significance                          272
Likely benign                                   216
Conflicting interpretations of pathogenicity     66
Benign                                           34
Benign/Likely benign                             23
Pathogenic                                       16
Likely pathogenic                                10
Pathogenic/Likely pathogenic                      3
Name: count, dtype: int64</code></pre>
</div>
</div>
<p>We can also look at the types of variants within gnomAD and their consequence, including variants that are not in ClinVar and have not been clinically annotated. This is the overall makeup of the entire gnomAD dataset:</p>
<div class="cell" data-execution_count="3">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">####</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ok now let me look at what other stuff we have to examine</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># df.columns</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>df[<span class="st">"VEP Annotation"</span>].value_counts()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="3">
<pre><code>VEP Annotation
intron_variant              1517
missense_variant            1141
synonymous_variant           411
splice_region_variant        191
frameshift_variant            57
stop_gained                   55
splice_donor_variant          32
3_prime_UTR_variant           24
splice_acceptor_variant       24
inframe_deletion              15
start_lost                     2
protein_altering_variant       1
Name: count, dtype: int64</code></pre>
</div>
</div>
<p>In this study, I propose to employ protein-level tools to predict deleteriousness. In this preliminary analysis the variants that are within gnomAD are paired with their predicted severity from the ESM1b model. We’ll only be able to examine the missense variants with this particular tool; let’s now see how many missense variants have been clinically annotated:</p>
<div class="cell" data-execution_count="4">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we can use missense with ESM1b -- some like frameshift, stop gained, um, inframe del, can be examined further or be presumed to be deleterious I'd think. n = 1 protein_altering, this may need to be treated manually the annotation is funky</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the annotation types to include (as of 20 Sep 2023, and for this analysis, just missense)</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>include_annotations <span class="op">=</span> [<span class="st">'missense_variant'</span>]</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>mask <span class="op">=</span> df[<span class="st">'VEP Annotation'</span>].isin(include_annotations)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>df_mis <span class="op">=</span> df[mask]</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># print(df_mis)</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>df_mis[<span class="st">'ClinVar Clinical Significance'</span>].value_counts(),</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="ch">\n</span><span class="st">Total of the gnomAD missense variants in ClinVar: "</span>,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>df_mis[<span class="st">'ClinVar Clinical Significance'</span>].value_counts().<span class="bu">sum</span>()</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>ClinVar Clinical Significance
Uncertain significance                          236
Conflicting interpretations of pathogenicity     23
Likely benign                                    19
Benign                                            7
Benign/Likely benign                              4
Likely pathogenic                                 2
Pathogenic                                        1
Name: count, dtype: int64 
Total of the gnomAD missense variants in ClinVar:  292</code></pre>
</div>
</div>
<p>There are 1141 missense mutants recorded in gnomAD. Of the missense mutations that are in Clinvar, the majority are of uncertain significance, 236/292. The ESM1b model has predictions for all possible missense mutations. The next couple code cells serve to get the same format between gnomAD data and ESM1b to describe variants.</p>
<p>(<em>And an aside</em>, the ESM1b model is also capable of accounting for stop gained and inframe deletions, of which there are 55 and 15 recorded in gnomAD; it requires more labor and the sequences themselves, though. There’s also all the intronic variants, these are outside of scope for now.)</p>
<p>Read in the data</p>
<div class="cell" data-execution_count="5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in the data</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>df_esm <span class="op">=</span> pd.read_csv(<span class="st">"data/ESM1b_SI.csv"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Define necessary functions</p>
<div class="cell" data-execution_count="6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define relevant functions/class for later use </span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> re</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> Bio.SeqUtils <span class="im">import</span> seq1, seq3</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="kw">class</span> seqConversion:</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="kw">def</span> <span class="fu">__init__</span>(<span class="va">self</span>) <span class="op">-&gt;</span> <span class="va">None</span>:</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>      <span class="cf">pass</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># intended to be private method, intermediate for convertGnomadHGVS below.      </span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="kw">def</span> _parseGnomadHGVS(<span class="va">self</span>, hgvsEntry):</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Define a regex pattern to match the format "stringNUMBERstring"</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    pattern <span class="op">=</span> <span class="vs">r'([A-Za-z]+)(\d+)([A-Za-z]+)'</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use re.match to find the first match in the input string, need to account for multiple and check the duplicates if exist</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>    match <span class="op">=</span> re.match(pattern, hgvsEntry)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> match: <span class="co"># which should be, but good in case weird data i don't anticipate</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Extract the three parts from the match object</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>        aa_wt <span class="op">=</span> match.group(<span class="dv">1</span>)</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>        pos <span class="op">=</span> match.group(<span class="dv">2</span>)</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>        aa_mut <span class="op">=</span> match.group(<span class="dv">3</span>)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Convert the numbers to an integer (if needed)</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>        pos <span class="op">=</span> <span class="bu">int</span>(pos)</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> aa_wt, pos, aa_mut</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span><span class="al">FIXME</span><span class="co">: May need more handling cases, so far this is working on SI data 22 Sep 2023</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Handle the case where no match is found</span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> <span class="va">None</span>, <span class="va">None</span>, <span class="va">None</span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>  <span class="kw">def</span> convertGnomadHGVS(<span class="va">self</span>, hgvsEntry):</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span><span class="al">NOTE</span><span class="co">: This assumes we're only passing in the protein-level variants, which, should be the only case this function is called, i.e. missense variants</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>    <span class="co"># strip 'p.'</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>    hgvsEntry <span class="op">=</span> hgvsEntry.lstrip(<span class="st">'p.'</span>)</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get the gnomad formatted variables</span></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>    aa_wt, pos, aa_mut <span class="op">=</span> <span class="va">self</span>._parseGnomadHGVS(hgvsEntry)</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> aa_wt <span class="kw">is</span> <span class="va">None</span> <span class="kw">or</span> pos <span class="kw">is</span> <span class="va">None</span> <span class="kw">or</span> aa_mut <span class="kw">is</span> <span class="va">None</span>:</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>            <span class="cf">return</span> <span class="va">None</span></span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>      <span class="co"># codes to one-letter codes; from BioSeq import above</span></span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>      aa_wt1 <span class="op">=</span> seq1(aa_wt)</span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>      aa_mut1 <span class="op">=</span> seq1(aa_mut)</span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>      <span class="co"># assemble the esm1b format of gnomad entry</span></span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>      newFormat <span class="op">=</span> <span class="ss">f"</span><span class="sc">{</span>aa_wt1<span class="sc">}{</span>pos<span class="sc">}{</span>aa_mut1<span class="sc">}</span><span class="ss">"</span></span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>      <span class="cf">return</span> newFormat</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>  <span class="kw">def</span> getProteinVariants(hgvsEntry):</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>    p_pattern <span class="op">=</span> <span class="vs">r"p.*"</span>  <span class="co"># Use .* to match zero or more occurrences of 'p'</span></span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>    matches <span class="op">=</span> re.findall(p_pattern, hgvsEntry)</span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> matches:</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> matches</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Handle the case where no match is found</span></span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> <span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We now finally have: - Only missense variants in this dataframe - Functions for converson in the class definition, used in Converter object - Now we can define new compatible format as a feature in gnomAD data, can cross reference gnomAD &lt;&gt; ESM1b model</p>
<p>So – define the 1-letter, ESM1b variant format as a new feature in gnomAD dataframe</p>
<div class="cell" data-execution_count="7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we defined this above, can do again for clarity</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Converter = seqConversion()</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># df_mis['HGVS_ESM1b']  = df_mis['HGVS Consequence'].apply(Converter.convertGnomadHGVS)</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># df_mis['HGVS_ESM1b'].head()</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>Converter <span class="op">=</span> seqConversion()</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co"># df_mis.loc[:, 'HGVS_ESM1b'] = df_mis['HGVS Consequence'].apply(Converter.convertGnomadHGVS)</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>df_mis <span class="op">=</span> df_mis.assign(HGVS_ESM1b<span class="op">=</span>df_mis[<span class="st">'HGVS Consequence'</span>].<span class="bu">apply</span>(Converter.convertGnomadHGVS))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Feature to enable cross referencing now defined.</p>
<p>So, now we can update our two dataframes, gnomAD and ESM1b. First, add the score feature into gnomAD.</p>
<div class="cell" data-execution_count="8">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge the 'score' feature from df_scores into df_mis based on matching 'variant' or 'HGVS_ESM1b' values</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>merged_df <span class="op">=</span> df_mis.merge(df_esm[[<span class="st">'variant'</span>, <span class="st">'score'</span>]], left_on<span class="op">=</span><span class="st">'HGVS_ESM1b'</span>, right_on<span class="op">=</span><span class="st">'variant'</span>, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># merged_df.head()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now let’s see the distribution of scores within gnomAD.</p>
<div class="cell" data-execution_count="9">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> plotly.express <span class="im">as</span> px</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the 'score' column from the merged DataFrame</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>scores <span class="op">=</span> merged_df[<span class="st">"score"</span>]</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> px.histogram(</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    merged_df,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    x<span class="op">=</span><span class="st">"score"</span>,</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    nbins<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>    opacity<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    labels<span class="op">=</span>{<span class="st">"score"</span>: <span class="st">"Scores"</span>},</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    title<span class="op">=</span><span class="st">"Distribution of Scores Within Gnomad"</span>,</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Customize the layout</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>fig.update_layout(bargap<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Show the plot</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>fig.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">

<div>                            <div id="2bb0a248-882f-4424-89f3-4a6927a5bf93" class="plotly-graph-div" style="height:525px; width:100%;"></div>            <script type="text/javascript">                require(["plotly"], function(Plotly) {                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById("2bb0a248-882f-4424-89f3-4a6927a5bf93")) {                    Plotly.newPlot(                        "2bb0a248-882f-4424-89f3-4a6927a5bf93",                        [{"alignmentgroup":"True","bingroup":"x","hovertemplate":"Scores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"","marker":{"color":"#636efa","opacity":1,"pattern":{"shape":""}},"name":"","nbinsx":20,"offsetgroup":"","orientation":"v","showlegend":false,"x":[-2.357,-3.158,-4.747,-6.465,-4.411,-2.642,-2.574,-4.16,-2.664,-5.22,-4.315,-2.647,-4.838,-5.463,-3.605,-5.089,-1.802,-6.435,-5.614,-7.055,-3.803,-3.651,-2.097,-2.097,-9.24,-8.665,-2.583,-2.917,-4.817,-4.282,-1.665,-4.14,-7.574,-7.917,-5.018,-5.784,-1.888,-8.163,-3.553,-4.4,-13.305,-4.379,-5.445,-5.233,-1.861,-3.652,-10.512,-7.916,-8.534,-4.29,-2.927,-5.93,-3.845,-7.886,-5.32,-6.68,-7.299,-9.789,-15.265,-10.476,-11.891,-5.275,-4.09,-8.499,-2.355,-5.678,-5.38,-14.373,-8.062,-7.243,-6.873,-8.676,-3.736,-9.04,-9.708,-3.696,-10.182,-0.588,-9.738,-6.241,-8.31,-11.722,-15.247,-13.009,-4.025,-10.524,-11.179,-9.417,-9.76,-4.552,-7.067,-5.752,-5.66,-6.18,-4.738,-4.634,-6.893,-9.624,-8.668,-8.922,-10.988,-7.562,-5.901,-11.494,-3.792,-9.311,-2.504,-4.959,-11.546,-7.82,-10.55,-9.838,-9.247,-10.043,-8.456,-12.181,-12.991,-12.021,-4.465,-6.256,-9.904,-4.194,-2.136,-3.267,-7.034,-5.563,-11.098,-13.547,-9.939,-6.154,-5.586,-6.942,-6.252,-10.267,-12.241,-1.22,-14.01,-0.875,-8.335,-9.771,-8.273,-9.418,-11.672,-9.5,-8.853,-10.275,-6.669,-6.83,-7.4,-9.819,-4.799,-9.548,-6.071,-5.819,-11.077,-11.328,-4.875,-4.966,-9.742,-13.477,-10.657,-8.58,-8.015,-13.313,-11.903,-11.24,-12.293,-11.623,-12.76,-13.53,-3.801,-1.363,-13.247,-4.784,-4.141,-13.102,-12.073,-14.094,-8.465,-11.546,-11.198,-0.146,-8.086,-11.091,-9.862,-5.93,-13.533,-8.62,-5.355,-3.337,-11.427,-4.273,-4.295,-7.574,-11.599,-10.674,-6.691,-10.946,-7.104,-13.568,-8.637,-5.757,-8.05,-11.013,-14.583,-12.225,-14.027,-12.347,-10.452,-7.88,-6.103,-3.456,-12.914,-8.472,-8.889,-3.96,-11.494,-5.509,-5.597,-10.518,-6.458,-4.019,-9.659,-13.291,-11.674,-1.651,-4.164,-9.316,-4.958,-8.399,-6.279,-8.697,-1.174,-8.541,-9.049,-13.129,-5.266,-9.681,-3.31,-2.449,-3.437,-3.37,-5.424,-4.533,-3.726,-1.678,-11.629,-10.342,-3.703,-11.822,-10.11,-12.838,-13.534,-4.311,-4.311,-4.204,-4.875,-4.562,-7.662,-4.765,-10.334,-13.632,-12.706,-12.976,-5.602,-7.879,-11.443,-6.627,-5.602,-5.359,-6.283,-5.701,-6.106,-4.523,-6.974,-4.179,-6.777,-3.717,-6.039,-4.971,-8.365,-9.676,-2.089,-5.754,-3.693,-6.249,-6.26,-7.117,-4.421,-12.839,-11.058,-9.07,-8.027,-9.008,-8.0,-5.507,-13.131,-9.68,-7.016,-15.569,-11.647,-6.695,-8.785,-4.496,-8.965,-5.076,-7.046,-7.556,-5.269,-11.751,-6.813,-5.199,-8.08,-3.485,-9.641,-6.675,-2.514,-6.811,-5.671,-3.512,-12.069,-5.09,-9.275,-7.14,-6.537,-10.949,-10.87,-11.428,-12.238,-12.902,-6.629,-10.938,-11.083,-11.256,-7.578,-12.091,-8.489,-9.929,-2.284,-7.034,-8.817,-5.314,-5.665,-2.842,-4.654,-5.243,-1.464,-6.988,-15.482,-6.166,-6.357,-11.748,-11.981,-10.585,-10.848,-11.517,-11.661,-9.824,-2.104,-11.947,-6.272,-3.38,-5.611,-1.82,-4.786,-5.322,-10.392,-6.744,-9.367,-11.271,-8.422,-11.223,-8.817,-8.737,-10.995,-11.932,0.526,-10.094,-7.498,-11.256,-11.52,-9.854,-5.663,-10.727,-13.123,-6.911,-11.193,-10.97,-13.024,-12.188,-14.2,-5.276,-8.416,-8.411,-0.717,-6.373,-7.704,-6.421,-6.996,-11.374,-12.9,-13.95,-12.562,-7.492,-7.161,-5.755,-9.453,-8.363,-7.852,-10.377,-7.991,-10.563,-10.447,-8.282,-9.9,-7.508,-6.713,-7.762,-8.468,-4.137,-6.684,-11.303,-0.094,-5.873,-9.586,-12.56,-3.356,-9.487,-6.721,-13.105,-10.482,-11.55,-4.742,-7.346,-6.596,-2.28,-4.499,-2.962,-3.724,-9.091,-6.978,-8.725,-9.726,-12.158,-9.024,-10.663,-10.15,-10.273,-5.262,-5.534,-7.145,-6.563,-4.352,-4.354,-5.935,-2.218,-5.439,-4.415,-2.322,-5.788,-9.522,-5.223,-8.242,-7.635,-5.672,-9.373,-9.153,-7.346,-7.676,-13.393,-9.851,-4.779,-12.62,-12.029,-8.022,-12.013,-8.943,-7.658,-6.88,-11.552,-6.611,-5.39,-7.396,-14.427,-9.083,-11.157,-6.468,-3.656,-6.418,-6.574,-2.965,-3.015,-5.71,-4.362,-5.081,-5.267,-8.77,-7.018,-10.136,-8.306,-4.196,-8.273,-7.741,-3.973,-8.565,-1.552,-11.868,-3.563,-7.17,-1.729,-2.457,-3.222,-2.452,-7.154,-8.426,-4.267,-3.873,-6.937,-6.992,-2.751,-7.655,-3.347,-5.54,-1.597,-9.3,-8.171,-11.548,-7.439,-8.497,-2.21,-4.767,-1.746,-7.752,-10.428,-0.606,-1.801,-3.403,-11.863,-4.553,-7.251,-2.248,-7.893,-3.976,-4.442,-6.106,-1.093,-0.167,-3.724,-8.689,-6.691,-3.473,-7.054,-6.322,-4.187,-3.267,-5.382,-2.495,-2.241,-5.049,-2.986,-3.132,-5.834,-5.844,-4.816,-6.279,-3.973,-2.651,-6.652,-5.146,-6.35,-4.545,-4.539,-3.794,-3.994,-6.792,-5.987,-4.19,0.054,-4.581,-3.823,-5.998,-8.848,-7.854,-4.931,-3.984,-4.207,-2.653,-3.831,-4.251,-4.187,-4.071,-5.279,-6.081,-7.726,-3.772,-7.484,-5.401,-5.58,-3.865,-2.766,-5.362,-3.8,-9.012,-8.291,-8.101,-13.475,-3.162,-11.549,-3.643,-8.22,-8.335,-9.116,-8.627,-11.381,-12.17,-12.786,-6.425,-8.47,-10.559,-11.542,-3.848,-12.716,-3.026,-11.963,-14.742,-10.615,-8.785,-11.04,-7.967,-11.117,-14.417,-6.827,-13.017,-9.716,-2.627,-7.294,-8.779,-8.263,-5.5,-7.478,-8.536,-7.362,-2.005,-5.417,-6.051,-4.526,-1.816,-7.22,-9.33,-6.877,-10.854,-4.739,-5.897,-8.052,-9.6,-7.557,-5.145,-8.242,-4.525,-9.889,-11.968,-9.061,-12.852,-12.468,-7.641,-6.409,-8.399,-5.987,-5.765,-6.92,-5.404,-3.839,-3.858,-6.014,-10.492,-9.247,-8.812,-13.295,-9.022,-7.164,-4.708,-9.792,-12.924,-11.328,-10.777,-7.373,-9.018,-12.728,-5.937,-12.849,-6.735,-8.444,-9.314,-11.753,-6.909,-5.905,-4.278,-10.268,-10.353,-11.109,-6.676,-6.482,-6.562,-5.747,-1.396,-3.534,-11.36,-9.486,-9.201,-9.036,-10.871,-11.589,-15.203,-12.431,-11.373,-10.986,-9.601,-6.925,-3.646,-4.292,-5.696,-9.692,-10.412,-10.756,-6.021,-13.468,-8.088,-3.829,-5.116,-16.026,-11.124,-12.054,-5.996,-5.368,-10.46,-13.77,-10.232,-10.313,-9.186,-9.476,-7.375,-5.905,-7.147,-11.638,-6.475,-12.864,-11.669,-12.337,-7.083,-5.955,-7.901,-10.341,-13.021,-5.366,-5.982,-7.109,-9.337,-12.035,-3.924,-8.054,-3.291,-3.795,-5.17,-5.841,-9.233,-6.923,-7.003,-9.602,-14.151,-7.12,-8.801,-14.893,-4.069,-10.118,-11.75,-7.229,-12.047,-6.608,-11.265,-8.357,-11.6,-9.324,-12.087,-9.151,-9.162,-11.664,-6.3,-11.478,-7.558,-5.75,-6.932,-9.954,-3.123,-6.871,-3.221,-8.792,-11.965,-7.806,-7.372,-11.3,-10.734,-10.615,-11.081,-3.332,-8.804,-4.215,-4.068,-5.003,-11.892,-10.28,-12.073,-7.619,-6.621,-12.423,-13.225,-11.773,-9.801,-13.359,-5.759,-3.156,-4.175,-5.725,-12.057,-4.868,-12.298,-6.811,-7.669,-6.896,-10.797,-9.935,-4.417,-7.86,-8.258,-2.901,-10.632,-7.467,-1.607,-5.213,-4.804,-2.793,-7.874,-10.559,-10.22,-6.332,-4.294,-2.269,-5.054,-13.695,-14.58,-7.88,-6.81,-12.974,-9.019,-6.035,-4.465,-6.113,-6.537,-3.416,-7.772,-2.261,-5.855,-5.193,-1.649,-5.12,-2.342,-8.035,-11.006,-4.617,-13.892,-8.158,-5.331,-10.313,-7.88,-6.92,-2.36,-10.531,-3.589,-7.287,-3.878,-4.711,-5.117,-13.687,-4.505,-11.658,-10.563,-11.504,-8.949,-11.694,-11.627,-9.643,-10.4,-5.835,-7.19,-4.085,-9.002,-8.994,-5.859,-5.576,-8.427,-11.391,-6.848,-4.881,-3.961,-2.24,-5.296,-6.853,-13.092,-11.536,-6.002,-8.956,-8.007,-11.63,-10.165,-7.669,-6.846,-8.013,-6.879,-6.404,-8.974,-6.851,-10.559,-7.163,-6.631,-9.658,-9.173,-8.037,-10.074,-11.076,-3.082,-7.602,-13.272,-7.452,-12.432,-10.839,-1.704,-10.279,-12.282,-7.0,-2.196,-4.842,-5.352,-2.772,-9.286,-7.954,-13.656,-10.33,-12.237,-6.514,-5.483,-10.782,-14.762,-9.651,-4.052,-8.489,-9.816,-0.656,-9.398,-7.686,-11.034,-5.49,-9.913,-12.377,-7.766,-4.172,-4.032,-4.402,-2.645,-3.587,-7.89,-9.394,-11.12,-3.967,-6.08,-3.788,-7.087,-8.22,-11.2,-10.235,-9.426,-8.413,-11.158,-11.265,-7.091,-3.307,-10.881,-12.403,-9.493,-3.905,-4.278,-8.772,-10.519,-9.713,-8.218,-9.031,-5.457,-6.417,-7.354,-14.119,-12.776,-7.759,-12.196,-11.898,-13.732,-11.867,-9.465,-5.021,-7.147,-8.643,-3.309,-6.034,-6.501,-4.932,-13.06,-3.774,-4.067,-10.139,-4.735,-5.743,-7.957,-1.885,-7.779,-5.563,-3.428,-6.225,-4.056,-7.377,-7.833,-11.409,-10.97,-8.151,-7.811,-3.661,-4.368,-3.474,-5.738,-5.712,-4.933,-3.029,-3.438,-6.656,-5.117,-9.097,-8.876,-11.577,-10.876,-2.039,-5.541,-4.755,-6.711,-7.294,-7.663,-10.808,-10.188,-9.538,-11.336,-11.228,-4.084,-6.396,-0.409,-4.286,-2.781,-5.03,-8.045,-5.652,-6.609,-6.977,-7.094,-3.833,-10.217,-0.032,-6.704,-6.847,-7.857,-1.923,-5.294,-7.273,-6.088,-4.102,-3.975,-4.164,-9.734,-9.506,-3.318,-4.576,-9.528,-2.676,-6.022,-5.601,-10.196,-5.643,-5.525,-3.023,-6.079,-4.865,-3.99,-4.722,-5.823,-3.726,-5.238,-3.963,-2.898,-3.349,-5.426,-5.61,-5.778,-4.291,-7.498,-0.102,-4.85,-6.436,-2.619,-3.935,-5.406,-4.897,-3.092,-2.964,-5.346,-3.307,-5.7,-3.231,-6.14,-6.713,-5.691,-12.899,-7.045,-5.403,-4.323,-5.201,-6.813,-3.955,-2.163,-10.206,-2.724,-4.913,-6.316,-8.397,-16.797,-8.858,-7.303,-6.888,-6.332],"xaxis":"x","yaxis":"y","type":"histogram"}],                        {"template":{"data":{"histogram2dcontour":[{"type":"histogram2dcontour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"choropleth":[{"type":"choropleth","colorbar":{"outlinewidth":0,"ticks":""}}],"histogram2d":[{"type":"histogram2d","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmap":[{"type":"heatmap","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmapgl":[{"type":"heatmapgl","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"contourcarpet":[{"type":"contourcarpet","colorbar":{"outlinewidth":0,"ticks":""}}],"contour":[{"type":"contour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"surface":[{"type":"surface","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"mesh3d":[{"type":"mesh3d","colorbar":{"outlinewidth":0,"ticks":""}}],"scatter":[{"fillpattern":{"fillmode":"overlay","size":10,"solidity":0.2},"type":"scatter"}],"parcoords":[{"type":"parcoords","line":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolargl":[{"type":"scatterpolargl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"bar":[{"error_x":{"color":"#2a3f5f"},"error_y":{"color":"#2a3f5f"},"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"bar"}],"scattergeo":[{"type":"scattergeo","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolar":[{"type":"scatterpolar","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"histogram":[{"marker":{"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"histogram"}],"scattergl":[{"type":"scattergl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatter3d":[{"type":"scatter3d","line":{"colorbar":{"outlinewidth":0,"ticks":""}},"marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattermapbox":[{"type":"scattermapbox","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterternary":[{"type":"scatterternary","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattercarpet":[{"type":"scattercarpet","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"carpet":[{"aaxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"baxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"type":"carpet"}],"table":[{"cells":{"fill":{"color":"#EBF0F8"},"line":{"color":"white"}},"header":{"fill":{"color":"#C8D4E3"},"line":{"color":"white"}},"type":"table"}],"barpolar":[{"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"barpolar"}],"pie":[{"automargin":true,"type":"pie"}]},"layout":{"autotypenumbers":"strict","colorway":["#636efa","#EF553B","#00cc96","#ab63fa","#FFA15A","#19d3f3","#FF6692","#B6E880","#FF97FF","#FECB52"],"font":{"color":"#2a3f5f"},"hovermode":"closest","hoverlabel":{"align":"left"},"paper_bgcolor":"white","plot_bgcolor":"#E5ECF6","polar":{"bgcolor":"#E5ECF6","angularaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"radialaxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"ternary":{"bgcolor":"#E5ECF6","aaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"baxis":{"gridcolor":"white","linecolor":"white","ticks":""},"caxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"coloraxis":{"colorbar":{"outlinewidth":0,"ticks":""}},"colorscale":{"sequential":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"sequentialminus":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"diverging":[[0,"#8e0152"],[0.1,"#c51b7d"],[0.2,"#de77ae"],[0.3,"#f1b6da"],[0.4,"#fde0ef"],[0.5,"#f7f7f7"],[0.6,"#e6f5d0"],[0.7,"#b8e186"],[0.8,"#7fbc41"],[0.9,"#4d9221"],[1,"#276419"]]},"xaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"yaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"scene":{"xaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"yaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"zaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2}},"shapedefaults":{"line":{"color":"#2a3f5f"}},"annotationdefaults":{"arrowcolor":"#2a3f5f","arrowhead":0,"arrowwidth":1},"geo":{"bgcolor":"white","landcolor":"#E5ECF6","subunitcolor":"white","showland":true,"showlakes":true,"lakecolor":"white"},"title":{"x":0.05},"mapbox":{"style":"light"}}},"xaxis":{"anchor":"y","domain":[0.0,1.0],"title":{"text":"Scores"}},"yaxis":{"anchor":"x","domain":[0.0,1.0],"title":{"text":"count"}},"legend":{"tracegroupgap":0},"title":{"text":"Distribution of Scores Within Gnomad"},"barmode":"relative","bargap":0.1},                        {"responsive": true}                    ).then(function(){
                            
var gd = document.getElementById('2bb0a248-882f-4424-89f3-4a6927a5bf93');
var x = new MutationObserver(function (mutations, observer) {{
        var display = window.getComputedStyle(gd).display;
        if (!display || display === 'none') {{
            console.log([gd, 'removed!']);
            Plotly.purge(gd);
            observer.disconnect();
        }}
}});

// Listen for the removal of the full notebook cells
var notebookContainer = gd.closest('#notebook-container');
if (notebookContainer) {{
    x.observe(notebookContainer, {childList: true});
}}

// Listen for the clearing of the current output cell
var outputEl = gd.closest('.output');
if (outputEl) {{
    x.observe(outputEl, {childList: true});
}}

                        })                };                });            </script>        </div>
</div>
</div>
<p>In their paper, the authors of the model found good separation at a threshold of -7.5:</p>
<div class="cell" data-execution_count="10">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of entries with a score below -15</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>count_below_minus_75 <span class="op">=</span> (merged_df[<span class="st">'score'</span>] <span class="op">&lt;</span> <span class="op">-</span><span class="fl">7.5</span>).<span class="bu">sum</span>()</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the count</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f'Number of entries with a score below -7.5: </span><span class="sc">{</span>count_below_minus_75<span class="sc">}</span><span class="ss">'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of entries with a score below -7.5: 531</code></pre>
</div>
</div>
<p>For the missense variants in gnomAD, this gives a predicted 531 deleterious variants. This is far more numerous than the current number of known pathogenic variants, and only a few known missense variants can be used to compare against this prediction. The below two histograms will attempt to gage how useful this threshold may actually be in this scenario.</p>
<p>And again, this is only for missense mutations.</p>
<p>The below shows the <em>three</em> known/likely MISSENSE mutations that are deleterious</p>
<div class="cell" data-execution_count="11">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># only got 3 that are missense, 23 conflicting. versus...</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>merged_df[<span class="st">'ClinVar Clinical Significance'</span>].value_counts()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="11">
<pre><code>ClinVar Clinical Significance
Uncertain significance                          236
Conflicting interpretations of pathogenicity     23
Likely benign                                    19
Benign                                            7
Benign/Likely benign                              4
Likely pathogenic                                 2
Pathogenic                                        1
Name: count, dtype: int64</code></pre>
</div>
</div>
<p>And against all variants in ClinVar:</p>
<div class="cell" data-execution_count="12">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 29, with 66 conflicting</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'ClinVar Clinical Significance'</span>].value_counts()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="12">
<pre><code>ClinVar Clinical Significance
Uncertain significance                          272
Likely benign                                   216
Conflicting interpretations of pathogenicity     66
Benign                                           34
Benign/Likely benign                             23
Pathogenic                                       16
Likely pathogenic                                10
Pathogenic/Likely pathogenic                      3
Name: count, dtype: int64</code></pre>
</div>
</div>
<p>And against the known/likely pathogenic variants in ClinVar, and see their cause:</p>
<div class="cell" data-execution_count="13">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>all_known_path <span class="op">=</span> df[<span class="st">'ClinVar Clinical Significance'</span>].isin([<span class="st">'Pathogenic'</span>, <span class="st">'Likely pathogenic'</span>, <span class="st">'Pathogenic/Likely pathogenic  '</span>])</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>akp_df <span class="op">=</span> df[all_known_path]</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>akp_df[<span class="st">'VEP Annotation'</span>].value_counts()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="13">
<pre><code>VEP Annotation
frameshift_variant         9
stop_gained                7
splice_donor_variant       5
missense_variant           3
splice_acceptor_variant    2
Name: count, dtype: int64</code></pre>
</div>
</div>
<p>29 pathogenic/likely pathogenic in ClinVar – these other variants, the 26 besides the 3 missense, are frameshit, stop gained and splice effect variants. The ESM1b model is capable of also examining stop gained and indel variants, but requires more intense analysis.</p>
<p>The next couple cells define and show dataframes for the pathogenic and benign variants to enable marking the figures. The authors of ESM1b note that a score threshold of -7.5 tends to give good separation between delterious and benign mutations.</p>
<p>Pathogenic missense dataframe:</p>
<div class="cell" data-execution_count="14">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the 'gnomad' DataFrame to select pathogenic variants</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>pathogenic_missense_variants <span class="op">=</span> merged_df[<span class="st">'ClinVar Clinical Significance'</span>].isin([<span class="st">'Pathogenic'</span>, <span class="st">'Likely pathogenic'</span>,]) <span class="co"># # adds 23 more samples that have varying scores</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>pmv_df <span class="op">=</span> merged_df[pathogenic_missense_variants]</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>pmv_df <span class="op">=</span> pmv_df.sort_values(by<span class="op">=</span><span class="st">'score'</span>, ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Only pathogenic, and likely pathogenic"</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>pmv_df[[<span class="st">'variant'</span>,<span class="st">'score'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Only pathogenic, and likely pathogenic</code></pre>
</div>
<div class="cell-output cell-output-display" data-execution_count="14">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">variant</th>
<th data-quarto-table-cell-role="th">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">356</td>
<td>C1229Y</td>
<td>-11.661</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">133</td>
<td>T1606I</td>
<td>-10.267</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">795</td>
<td>D536V</td>
<td>-8.792</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>Benign missense dataframe:</p>
<div class="cell" data-execution_count="15">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># merged_df['ClinVar Clinical Significance'].value_counts() # peek the categories</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter, get benign bariants</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>benign_missense_variants <span class="op">=</span> merged_df[<span class="st">'ClinVar Clinical Significance'</span>].isin([<span class="st">'Benign'</span>, <span class="st">'Benign/Likely benign'</span>, <span class="st">'Likely benign'</span>])</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="co"># bmv: benign missense variants</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>bmv_df <span class="op">=</span> merged_df[benign_missense_variants]</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>bmv_df <span class="op">=</span> bmv_df.sort_values(by<span class="op">=</span><span class="st">'score'</span>, ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Only benign, and likely benign"</span>)</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>bmv_df[[<span class="st">'variant'</span>,<span class="st">'score'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Only benign, and likely benign</code></pre>
</div>
<div class="cell-output cell-output-display" data-execution_count="15">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">variant</th>
<th data-quarto-table-cell-role="th">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">888</td>
<td>I389N</td>
<td>-11.627</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">875</td>
<td>G407R</td>
<td>-10.531</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1041</td>
<td>P173A</td>
<td>-9.097</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">496</td>
<td>Q1000P</td>
<td>-8.770</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1135</td>
<td>V15F</td>
<td>-8.397</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1072</td>
<td>T122A</td>
<td>-7.857</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">334</td>
<td>T1258A</td>
<td>-7.578</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">323</td>
<td>G1272D</td>
<td>-7.140</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">649</td>
<td>R774K</td>
<td>-6.877</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">367</td>
<td>P1200S</td>
<td>-6.744</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">549</td>
<td>L914I</td>
<td>-6.691</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">995</td>
<td>T231A</td>
<td>-6.417</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">395</td>
<td>E1158Q</td>
<td>-6.373</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">119</td>
<td>D1623A</td>
<td>-6.256</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">638</td>
<td>Y781F</td>
<td>-5.500</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">994</td>
<td>T231S</td>
<td>-5.457</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">555</td>
<td>F905C</td>
<td>-5.382</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">15</td>
<td>S1795L</td>
<td>-5.089</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">150</td>
<td>T1574S</td>
<td>-4.799</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">94</td>
<td>V1667I</td>
<td>-4.738</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">95</td>
<td>V1667L</td>
<td>-4.634</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">215</td>
<td>K1473R</td>
<td>-3.960</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">45</td>
<td>Q1749K</td>
<td>-3.652</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">8</td>
<td>R1808H</td>
<td>-2.664</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">11</td>
<td>T1802S</td>
<td>-2.647</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">239</td>
<td>E1414K</td>
<td>-2.449</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1019</td>
<td>S200G</td>
<td>-1.885</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">523</td>
<td>R963K</td>
<td>-1.597</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">135</td>
<td>A1602V</td>
<td>-1.220</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">181</td>
<td>M1523I</td>
<td>-0.146</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>With the very limited sample size of 3, all known/likely pathogenic missense variants’ score is below -7.5. And known/likely benign tend to be above -7.5. However, limited data precludes any real inference here.</p>
<p>Create the feature first, the plotly express API does not accept conditional statements for coloring.</p>
<div class="cell" data-execution_count="16">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the threshold</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>threshold <span class="op">=</span> <span class="op">-</span><span class="fl">7.5</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new feature 'below threshold' based on the 'score' column</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>merged_df[<span class="st">'below threshold'</span>] <span class="op">=</span> merged_df[<span class="st">'score'</span>].<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">"yes"</span> <span class="cf">if</span> x <span class="op">&lt;</span> threshold <span class="cf">else</span> <span class="st">"no"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>

<!-- 

::: {.cell execution_count=17}
``` {.python .cell-code}
# Extract the 'score' column from the merged DataFrame
scores = merged_df["score"]

fig = px.histogram(
    merged_df,
    x="score",
    color = "below threshold",
    nbins=20,
    opacity=1,
    labels={"score": "Scores"},
    title="Distribution of Scores Within Gnomad",
)

# Customize the layout
fig.update_layout(bargap=0.1)

# Show the plot
fig.show()
```

::: {.cell-output .cell-output-display}

```{=html}
<div>                            <div id="abf1aa72-90a5-4b9f-b7d4-a107bb186c34" class="plotly-graph-div" style="height:525px; width:100%;"></div>            <script type="text/javascript">                require(["plotly"], function(Plotly) {                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById("abf1aa72-90a5-4b9f-b7d4-a107bb186c34")) {                    Plotly.newPlot(                        "abf1aa72-90a5-4b9f-b7d4-a107bb186c34",                        [{"alignmentgroup":"True","bingroup":"x","hovertemplate":"below threshold=no\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"no","marker":{"color":"#636efa","opacity":1,"pattern":{"shape":""}},"name":"no","nbinsx":20,"offsetgroup":"no","orientation":"v","showlegend":true,"x":[-2.357,-3.158,-4.747,-6.465,-4.411,-2.642,-2.574,-4.16,-2.664,-5.22,-4.315,-2.647,-4.838,-5.463,-3.605,-5.089,-1.802,-6.435,-5.614,-7.055,-3.803,-3.651,-2.097,-2.097,-2.583,-2.917,-4.817,-4.282,-1.665,-4.14,-5.018,-5.784,-1.888,-3.553,-4.4,-4.379,-5.445,-5.233,-1.861,-3.652,-4.29,-2.927,-5.93,-3.845,-5.32,-6.68,-7.299,-5.275,-4.09,-2.355,-5.678,-5.38,-7.243,-6.873,-3.736,-3.696,-0.588,-6.241,-4.025,-4.552,-7.067,-5.752,-5.66,-6.18,-4.738,-4.634,-6.893,-5.901,-3.792,-2.504,-4.959,-4.465,-6.256,-4.194,-2.136,-3.267,-7.034,-5.563,-6.154,-5.586,-6.942,-6.252,-1.22,-0.875,-6.669,-6.83,-7.4,-4.799,-6.071,-5.819,-4.875,-4.966,-3.801,-1.363,-4.784,-4.141,-0.146,-5.93,-5.355,-3.337,-4.273,-4.295,-6.691,-7.104,-5.757,-6.103,-3.456,-3.96,-5.509,-5.597,-6.458,-4.019,-1.651,-4.164,-4.958,-6.279,-1.174,-5.266,-3.31,-2.449,-3.437,-3.37,-5.424,-4.533,-3.726,-1.678,-3.703,-4.311,-4.311,-4.204,-4.875,-4.562,-4.765,-5.602,-6.627,-5.602,-5.359,-6.283,-5.701,-6.106,-4.523,-6.974,-4.179,-6.777,-3.717,-6.039,-4.971,-2.089,-5.754,-3.693,-6.249,-6.26,-7.117,-4.421,-5.507,-7.016,-6.695,-4.496,-5.076,-7.046,-5.269,-6.813,-5.199,-3.485,-6.675,-2.514,-6.811,-5.671,-3.512,-5.09,-7.14,-6.537,-6.629,-2.284,-7.034,-5.314,-5.665,-2.842,-4.654,-5.243,-1.464,-6.988,-6.166,-6.357,-2.104,-6.272,-3.38,-5.611,-1.82,-4.786,-5.322,-6.744,0.526,-7.498,-5.663,-6.911,-5.276,-0.717,-6.373,-6.421,-6.996,-7.492,-7.161,-5.755,-6.713,-4.137,-6.684,-0.094,-5.873,-3.356,-6.721,-4.742,-7.346,-6.596,-2.28,-4.499,-2.962,-3.724,-6.978,-5.262,-5.534,-7.145,-6.563,-4.352,-4.354,-5.935,-2.218,-5.439,-4.415,-2.322,-5.788,-5.223,-5.672,-7.346,-4.779,-6.88,-6.611,-5.39,-7.396,-6.468,-3.656,-6.418,-6.574,-2.965,-3.015,-5.71,-4.362,-5.081,-5.267,-7.018,-4.196,-3.973,-1.552,-3.563,-7.17,-1.729,-2.457,-3.222,-2.452,-7.154,-4.267,-3.873,-6.937,-6.992,-2.751,-3.347,-5.54,-1.597,-7.439,-2.21,-4.767,-1.746,-0.606,-1.801,-3.403,-4.553,-7.251,-2.248,-3.976,-4.442,-6.106,-1.093,-0.167,-3.724,-6.691,-3.473,-7.054,-6.322,-4.187,-3.267,-5.382,-2.495,-2.241,-5.049,-2.986,-3.132,-5.834,-5.844,-4.816,-6.279,-3.973,-2.651,-6.652,-5.146,-6.35,-4.545,-4.539,-3.794,-3.994,-6.792,-5.987,-4.19,0.054,-4.581,-3.823,-5.998,-4.931,-3.984,-4.207,-2.653,-3.831,-4.251,-4.187,-4.071,-5.279,-6.081,-3.772,-7.484,-5.401,-5.58,-3.865,-2.766,-5.362,-3.8,-3.162,-3.643,-6.425,-3.848,-3.026,-6.827,-2.627,-7.294,-5.5,-7.478,-7.362,-2.005,-5.417,-6.051,-4.526,-1.816,-7.22,-6.877,-4.739,-5.897,-5.145,-4.525,-6.409,-5.987,-5.765,-6.92,-5.404,-3.839,-3.858,-6.014,-7.164,-4.708,-7.373,-5.937,-6.735,-6.909,-5.905,-4.278,-6.676,-6.482,-6.562,-5.747,-1.396,-3.534,-6.925,-3.646,-4.292,-5.696,-6.021,-3.829,-5.116,-5.996,-5.368,-7.375,-5.905,-7.147,-6.475,-7.083,-5.955,-5.366,-5.982,-7.109,-3.924,-3.291,-3.795,-5.17,-5.841,-6.923,-7.003,-7.12,-4.069,-7.229,-6.608,-6.3,-5.75,-6.932,-3.123,-6.871,-3.221,-7.372,-3.332,-4.215,-4.068,-5.003,-6.621,-5.759,-3.156,-4.175,-5.725,-4.868,-6.811,-6.896,-4.417,-2.901,-7.467,-1.607,-5.213,-4.804,-2.793,-6.332,-4.294,-2.269,-5.054,-6.81,-6.035,-4.465,-6.113,-6.537,-3.416,-2.261,-5.855,-5.193,-1.649,-5.12,-2.342,-4.617,-5.331,-6.92,-2.36,-3.589,-7.287,-3.878,-4.711,-5.117,-4.505,-5.835,-7.19,-4.085,-5.859,-5.576,-6.848,-4.881,-3.961,-2.24,-5.296,-6.853,-6.002,-6.846,-6.879,-6.404,-6.851,-7.163,-6.631,-3.082,-7.452,-1.704,-7.0,-2.196,-4.842,-5.352,-2.772,-6.514,-5.483,-4.052,-0.656,-5.49,-4.172,-4.032,-4.402,-2.645,-3.587,-3.967,-6.08,-3.788,-7.087,-7.091,-3.307,-3.905,-4.278,-5.457,-6.417,-7.354,-5.021,-7.147,-3.309,-6.034,-6.501,-4.932,-3.774,-4.067,-4.735,-5.743,-1.885,-5.563,-3.428,-6.225,-4.056,-7.377,-3.661,-4.368,-3.474,-5.738,-5.712,-4.933,-3.029,-3.438,-6.656,-5.117,-2.039,-5.541,-4.755,-6.711,-7.294,-4.084,-6.396,-0.409,-4.286,-2.781,-5.03,-5.652,-6.609,-6.977,-7.094,-3.833,-0.032,-6.704,-6.847,-1.923,-5.294,-7.273,-6.088,-4.102,-3.975,-4.164,-3.318,-4.576,-2.676,-6.022,-5.601,-5.643,-5.525,-3.023,-6.079,-4.865,-3.99,-4.722,-5.823,-3.726,-5.238,-3.963,-2.898,-3.349,-5.426,-5.61,-5.778,-4.291,-7.498,-0.102,-4.85,-6.436,-2.619,-3.935,-5.406,-4.897,-3.092,-2.964,-5.346,-3.307,-5.7,-3.231,-6.14,-6.713,-5.691,-7.045,-5.403,-4.323,-5.201,-6.813,-3.955,-2.163,-2.724,-4.913,-6.316,-7.303,-6.888,-6.332],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"below threshold=yes\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"yes","marker":{"color":"#EF553B","opacity":1,"pattern":{"shape":""}},"name":"yes","nbinsx":20,"offsetgroup":"yes","orientation":"v","showlegend":true,"x":[-9.24,-8.665,-7.574,-7.917,-8.163,-13.305,-10.512,-7.916,-8.534,-7.886,-9.789,-15.265,-10.476,-11.891,-8.499,-14.373,-8.062,-8.676,-9.04,-9.708,-10.182,-9.738,-8.31,-11.722,-15.247,-13.009,-10.524,-11.179,-9.417,-9.76,-9.624,-8.668,-8.922,-10.988,-7.562,-11.494,-9.311,-11.546,-7.82,-10.55,-9.838,-9.247,-10.043,-8.456,-12.181,-12.991,-12.021,-9.904,-11.098,-13.547,-9.939,-10.267,-12.241,-14.01,-8.335,-9.771,-8.273,-9.418,-11.672,-9.5,-8.853,-10.275,-9.819,-9.548,-11.077,-11.328,-9.742,-13.477,-10.657,-8.58,-8.015,-13.313,-11.903,-11.24,-12.293,-11.623,-12.76,-13.53,-13.247,-13.102,-12.073,-14.094,-8.465,-11.546,-11.198,-8.086,-11.091,-9.862,-13.533,-8.62,-11.427,-7.574,-11.599,-10.674,-10.946,-13.568,-8.637,-8.05,-11.013,-14.583,-12.225,-14.027,-12.347,-10.452,-7.88,-12.914,-8.472,-8.889,-11.494,-10.518,-9.659,-13.291,-11.674,-9.316,-8.399,-8.697,-8.541,-9.049,-13.129,-9.681,-11.629,-10.342,-11.822,-10.11,-12.838,-13.534,-7.662,-10.334,-13.632,-12.706,-12.976,-7.879,-11.443,-8.365,-9.676,-12.839,-11.058,-9.07,-8.027,-9.008,-8.0,-13.131,-9.68,-15.569,-11.647,-8.785,-8.965,-7.556,-11.751,-8.08,-9.641,-12.069,-9.275,-10.949,-10.87,-11.428,-12.238,-12.902,-10.938,-11.083,-11.256,-7.578,-12.091,-8.489,-9.929,-8.817,-15.482,-11.748,-11.981,-10.585,-10.848,-11.517,-11.661,-9.824,-11.947,-10.392,-9.367,-11.271,-8.422,-11.223,-8.817,-8.737,-10.995,-11.932,-10.094,-11.256,-11.52,-9.854,-10.727,-13.123,-11.193,-10.97,-13.024,-12.188,-14.2,-8.416,-8.411,-7.704,-11.374,-12.9,-13.95,-12.562,-9.453,-8.363,-7.852,-10.377,-7.991,-10.563,-10.447,-8.282,-9.9,-7.508,-7.762,-8.468,-11.303,-9.586,-12.56,-9.487,-13.105,-10.482,-11.55,-9.091,-8.725,-9.726,-12.158,-9.024,-10.663,-10.15,-10.273,-9.522,-8.242,-7.635,-9.373,-9.153,-7.676,-13.393,-9.851,-12.62,-12.029,-8.022,-12.013,-8.943,-7.658,-11.552,-14.427,-9.083,-11.157,-8.77,-10.136,-8.306,-8.273,-7.741,-8.565,-11.868,-8.426,-7.655,-9.3,-8.171,-11.548,-8.497,-7.752,-10.428,-11.863,-7.893,-8.689,-8.848,-7.854,-7.726,-9.012,-8.291,-8.101,-13.475,-11.549,-8.22,-8.335,-9.116,-8.627,-11.381,-12.17,-12.786,-8.47,-10.559,-11.542,-12.716,-11.963,-14.742,-10.615,-8.785,-11.04,-7.967,-11.117,-14.417,-13.017,-9.716,-8.779,-8.263,-8.536,-9.33,-10.854,-8.052,-9.6,-7.557,-8.242,-9.889,-11.968,-9.061,-12.852,-12.468,-7.641,-8.399,-10.492,-9.247,-8.812,-13.295,-9.022,-9.792,-12.924,-11.328,-10.777,-9.018,-12.728,-12.849,-8.444,-9.314,-11.753,-10.268,-10.353,-11.109,-11.36,-9.486,-9.201,-9.036,-10.871,-11.589,-15.203,-12.431,-11.373,-10.986,-9.601,-9.692,-10.412,-10.756,-13.468,-8.088,-16.026,-11.124,-12.054,-10.46,-13.77,-10.232,-10.313,-9.186,-9.476,-11.638,-12.864,-11.669,-12.337,-7.901,-10.341,-13.021,-9.337,-12.035,-8.054,-9.233,-9.602,-14.151,-8.801,-14.893,-10.118,-11.75,-12.047,-11.265,-8.357,-11.6,-9.324,-12.087,-9.151,-9.162,-11.664,-11.478,-7.558,-9.954,-8.792,-11.965,-7.806,-11.3,-10.734,-10.615,-11.081,-8.804,-11.892,-10.28,-12.073,-7.619,-12.423,-13.225,-11.773,-9.801,-13.359,-12.057,-12.298,-7.669,-10.797,-9.935,-7.86,-8.258,-10.632,-7.874,-10.559,-10.22,-13.695,-14.58,-7.88,-12.974,-9.019,-7.772,-8.035,-11.006,-13.892,-8.158,-10.313,-7.88,-10.531,-13.687,-11.658,-10.563,-11.504,-8.949,-11.694,-11.627,-9.643,-10.4,-9.002,-8.994,-8.427,-11.391,-13.092,-11.536,-8.956,-8.007,-11.63,-10.165,-7.669,-8.013,-8.974,-10.559,-9.658,-9.173,-8.037,-10.074,-11.076,-7.602,-13.272,-12.432,-10.839,-10.279,-12.282,-9.286,-7.954,-13.656,-10.33,-12.237,-10.782,-14.762,-9.651,-8.489,-9.816,-9.398,-7.686,-11.034,-9.913,-12.377,-7.766,-7.89,-9.394,-11.12,-8.22,-11.2,-10.235,-9.426,-8.413,-11.158,-11.265,-10.881,-12.403,-9.493,-8.772,-10.519,-9.713,-8.218,-9.031,-14.119,-12.776,-7.759,-12.196,-11.898,-13.732,-11.867,-9.465,-8.643,-13.06,-10.139,-7.957,-7.779,-7.833,-11.409,-10.97,-8.151,-7.811,-9.097,-8.876,-11.577,-10.876,-7.663,-10.808,-10.188,-9.538,-11.336,-11.228,-8.045,-10.217,-7.857,-9.734,-9.506,-9.528,-10.196,-12.899,-10.206,-8.397,-16.797,-8.858],"xaxis":"x","yaxis":"y","type":"histogram"}],                        {"template":{"data":{"histogram2dcontour":[{"type":"histogram2dcontour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"choropleth":[{"type":"choropleth","colorbar":{"outlinewidth":0,"ticks":""}}],"histogram2d":[{"type":"histogram2d","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmap":[{"type":"heatmap","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmapgl":[{"type":"heatmapgl","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"contourcarpet":[{"type":"contourcarpet","colorbar":{"outlinewidth":0,"ticks":""}}],"contour":[{"type":"contour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"surface":[{"type":"surface","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"mesh3d":[{"type":"mesh3d","colorbar":{"outlinewidth":0,"ticks":""}}],"scatter":[{"fillpattern":{"fillmode":"overlay","size":10,"solidity":0.2},"type":"scatter"}],"parcoords":[{"type":"parcoords","line":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolargl":[{"type":"scatterpolargl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"bar":[{"error_x":{"color":"#2a3f5f"},"error_y":{"color":"#2a3f5f"},"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"bar"}],"scattergeo":[{"type":"scattergeo","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolar":[{"type":"scatterpolar","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"histogram":[{"marker":{"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"histogram"}],"scattergl":[{"type":"scattergl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatter3d":[{"type":"scatter3d","line":{"colorbar":{"outlinewidth":0,"ticks":""}},"marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattermapbox":[{"type":"scattermapbox","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterternary":[{"type":"scatterternary","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattercarpet":[{"type":"scattercarpet","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"carpet":[{"aaxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"baxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"type":"carpet"}],"table":[{"cells":{"fill":{"color":"#EBF0F8"},"line":{"color":"white"}},"header":{"fill":{"color":"#C8D4E3"},"line":{"color":"white"}},"type":"table"}],"barpolar":[{"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"barpolar"}],"pie":[{"automargin":true,"type":"pie"}]},"layout":{"autotypenumbers":"strict","colorway":["#636efa","#EF553B","#00cc96","#ab63fa","#FFA15A","#19d3f3","#FF6692","#B6E880","#FF97FF","#FECB52"],"font":{"color":"#2a3f5f"},"hovermode":"closest","hoverlabel":{"align":"left"},"paper_bgcolor":"white","plot_bgcolor":"#E5ECF6","polar":{"bgcolor":"#E5ECF6","angularaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"radialaxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"ternary":{"bgcolor":"#E5ECF6","aaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"baxis":{"gridcolor":"white","linecolor":"white","ticks":""},"caxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"coloraxis":{"colorbar":{"outlinewidth":0,"ticks":""}},"colorscale":{"sequential":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"sequentialminus":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"diverging":[[0,"#8e0152"],[0.1,"#c51b7d"],[0.2,"#de77ae"],[0.3,"#f1b6da"],[0.4,"#fde0ef"],[0.5,"#f7f7f7"],[0.6,"#e6f5d0"],[0.7,"#b8e186"],[0.8,"#7fbc41"],[0.9,"#4d9221"],[1,"#276419"]]},"xaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"yaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"scene":{"xaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"yaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"zaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2}},"shapedefaults":{"line":{"color":"#2a3f5f"}},"annotationdefaults":{"arrowcolor":"#2a3f5f","arrowhead":0,"arrowwidth":1},"geo":{"bgcolor":"white","landcolor":"#E5ECF6","subunitcolor":"white","showland":true,"showlakes":true,"lakecolor":"white"},"title":{"x":0.05},"mapbox":{"style":"light"}}},"xaxis":{"anchor":"y","domain":[0.0,1.0],"title":{"text":"Scores"}},"yaxis":{"anchor":"x","domain":[0.0,1.0],"title":{"text":"count"}},"legend":{"title":{"text":"below threshold"},"tracegroupgap":0},"title":{"text":"Distribution of Scores Within Gnomad"},"barmode":"relative","bargap":0.1},                        {"responsive": true}                    ).then(function(){
                            
var gd = document.getElementById('abf1aa72-90a5-4b9f-b7d4-a107bb186c34');
var x = new MutationObserver(function (mutations, observer) {{
        var display = window.getComputedStyle(gd).display;
        if (!display || display === 'none') {{
            console.log([gd, 'removed!']);
            Plotly.purge(gd);
            observer.disconnect();
        }}
}});

// Listen for the removal of the full notebook cells
var notebookContainer = gd.closest('#notebook-container');
if (notebookContainer) {{
    x.observe(notebookContainer, {childList: true});
}}

// Listen for the clearing of the current output cell
var outputEl = gd.closest('.output');
if (outputEl) {{
    x.observe(outputEl, {childList: true});
}}

                        })                };                });            </script>        </div>
```

:::
:::


... Lot of variants lol

Next plot overlays the benign/likely benign and pathogenic/likely pathogenic; green as benign, red as pathogenic. -->
<div class="cell" data-execution_count="18">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import matplotlib.pyplot as plt</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> plotly.express <span class="im">as</span> px</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> px.histogram(</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    merged_df,</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>    x<span class="op">=</span><span class="st">"score"</span>,</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    color <span class="op">=</span> <span class="st">"below threshold"</span>,</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>    nbins<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>    opacity<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>    labels<span class="op">=</span>{<span class="st">"score"</span>: <span class="st">"Scores"</span>},</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>    title<span class="op">=</span><span class="st">"Distribution of Scores Within Gnomad"</span>,</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add markers or lines for known pathogenic variants</span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx, row <span class="kw">in</span> pmv_df.iterrows():</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>    fig.add_shape(<span class="bu">type</span><span class="op">=</span><span class="st">'line'</span>, x0<span class="op">=</span>row[<span class="st">'score'</span>], x1<span class="op">=</span>row[<span class="st">'score'</span>], y0<span class="op">=</span><span class="dv">0</span>, y1<span class="op">=</span><span class="dv">1</span>, opacity <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>                  xref<span class="op">=</span><span class="st">'x'</span>, yref<span class="op">=</span><span class="st">'paper'</span>, line<span class="op">=</span><span class="bu">dict</span>(color<span class="op">=</span><span class="st">'red'</span>))</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Add markers/lines for benign variants</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx, row <span class="kw">in</span> bmv_df.iterrows():</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>    fig.add_shape(<span class="bu">type</span><span class="op">=</span><span class="st">'line'</span>, x0<span class="op">=</span>row[<span class="st">'score'</span>], x1<span class="op">=</span>row[<span class="st">'score'</span>], y0<span class="op">=</span><span class="dv">0</span>, y1<span class="op">=</span><span class="dv">1</span>, opacity <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>                  xref<span class="op">=</span><span class="st">'x'</span>, yref<span class="op">=</span><span class="st">'paper'</span>, line<span class="op">=</span><span class="bu">dict</span>(color<span class="op">=</span><span class="st">'green'</span>))</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Customize the layout</span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>fig.update_layout(bargap<span class="op">=</span><span class="fl">0.1</span>)</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Show the plot</span></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>fig.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">

<div>                            <div id="49b4255d-0485-43ad-bc98-bc33bb1320da" class="plotly-graph-div" style="height:525px; width:100%;"></div>            <script type="text/javascript">                require(["plotly"], function(Plotly) {                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById("49b4255d-0485-43ad-bc98-bc33bb1320da")) {                    Plotly.newPlot(                        "49b4255d-0485-43ad-bc98-bc33bb1320da",                        [{"alignmentgroup":"True","bingroup":"x","hovertemplate":"below threshold=no\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"no","marker":{"color":"#636efa","opacity":1,"pattern":{"shape":""}},"name":"no","nbinsx":20,"offsetgroup":"no","orientation":"v","showlegend":true,"x":[-2.357,-3.158,-4.747,-6.465,-4.411,-2.642,-2.574,-4.16,-2.664,-5.22,-4.315,-2.647,-4.838,-5.463,-3.605,-5.089,-1.802,-6.435,-5.614,-7.055,-3.803,-3.651,-2.097,-2.097,-2.583,-2.917,-4.817,-4.282,-1.665,-4.14,-5.018,-5.784,-1.888,-3.553,-4.4,-4.379,-5.445,-5.233,-1.861,-3.652,-4.29,-2.927,-5.93,-3.845,-5.32,-6.68,-7.299,-5.275,-4.09,-2.355,-5.678,-5.38,-7.243,-6.873,-3.736,-3.696,-0.588,-6.241,-4.025,-4.552,-7.067,-5.752,-5.66,-6.18,-4.738,-4.634,-6.893,-5.901,-3.792,-2.504,-4.959,-4.465,-6.256,-4.194,-2.136,-3.267,-7.034,-5.563,-6.154,-5.586,-6.942,-6.252,-1.22,-0.875,-6.669,-6.83,-7.4,-4.799,-6.071,-5.819,-4.875,-4.966,-3.801,-1.363,-4.784,-4.141,-0.146,-5.93,-5.355,-3.337,-4.273,-4.295,-6.691,-7.104,-5.757,-6.103,-3.456,-3.96,-5.509,-5.597,-6.458,-4.019,-1.651,-4.164,-4.958,-6.279,-1.174,-5.266,-3.31,-2.449,-3.437,-3.37,-5.424,-4.533,-3.726,-1.678,-3.703,-4.311,-4.311,-4.204,-4.875,-4.562,-4.765,-5.602,-6.627,-5.602,-5.359,-6.283,-5.701,-6.106,-4.523,-6.974,-4.179,-6.777,-3.717,-6.039,-4.971,-2.089,-5.754,-3.693,-6.249,-6.26,-7.117,-4.421,-5.507,-7.016,-6.695,-4.496,-5.076,-7.046,-5.269,-6.813,-5.199,-3.485,-6.675,-2.514,-6.811,-5.671,-3.512,-5.09,-7.14,-6.537,-6.629,-2.284,-7.034,-5.314,-5.665,-2.842,-4.654,-5.243,-1.464,-6.988,-6.166,-6.357,-2.104,-6.272,-3.38,-5.611,-1.82,-4.786,-5.322,-6.744,0.526,-7.498,-5.663,-6.911,-5.276,-0.717,-6.373,-6.421,-6.996,-7.492,-7.161,-5.755,-6.713,-4.137,-6.684,-0.094,-5.873,-3.356,-6.721,-4.742,-7.346,-6.596,-2.28,-4.499,-2.962,-3.724,-6.978,-5.262,-5.534,-7.145,-6.563,-4.352,-4.354,-5.935,-2.218,-5.439,-4.415,-2.322,-5.788,-5.223,-5.672,-7.346,-4.779,-6.88,-6.611,-5.39,-7.396,-6.468,-3.656,-6.418,-6.574,-2.965,-3.015,-5.71,-4.362,-5.081,-5.267,-7.018,-4.196,-3.973,-1.552,-3.563,-7.17,-1.729,-2.457,-3.222,-2.452,-7.154,-4.267,-3.873,-6.937,-6.992,-2.751,-3.347,-5.54,-1.597,-7.439,-2.21,-4.767,-1.746,-0.606,-1.801,-3.403,-4.553,-7.251,-2.248,-3.976,-4.442,-6.106,-1.093,-0.167,-3.724,-6.691,-3.473,-7.054,-6.322,-4.187,-3.267,-5.382,-2.495,-2.241,-5.049,-2.986,-3.132,-5.834,-5.844,-4.816,-6.279,-3.973,-2.651,-6.652,-5.146,-6.35,-4.545,-4.539,-3.794,-3.994,-6.792,-5.987,-4.19,0.054,-4.581,-3.823,-5.998,-4.931,-3.984,-4.207,-2.653,-3.831,-4.251,-4.187,-4.071,-5.279,-6.081,-3.772,-7.484,-5.401,-5.58,-3.865,-2.766,-5.362,-3.8,-3.162,-3.643,-6.425,-3.848,-3.026,-6.827,-2.627,-7.294,-5.5,-7.478,-7.362,-2.005,-5.417,-6.051,-4.526,-1.816,-7.22,-6.877,-4.739,-5.897,-5.145,-4.525,-6.409,-5.987,-5.765,-6.92,-5.404,-3.839,-3.858,-6.014,-7.164,-4.708,-7.373,-5.937,-6.735,-6.909,-5.905,-4.278,-6.676,-6.482,-6.562,-5.747,-1.396,-3.534,-6.925,-3.646,-4.292,-5.696,-6.021,-3.829,-5.116,-5.996,-5.368,-7.375,-5.905,-7.147,-6.475,-7.083,-5.955,-5.366,-5.982,-7.109,-3.924,-3.291,-3.795,-5.17,-5.841,-6.923,-7.003,-7.12,-4.069,-7.229,-6.608,-6.3,-5.75,-6.932,-3.123,-6.871,-3.221,-7.372,-3.332,-4.215,-4.068,-5.003,-6.621,-5.759,-3.156,-4.175,-5.725,-4.868,-6.811,-6.896,-4.417,-2.901,-7.467,-1.607,-5.213,-4.804,-2.793,-6.332,-4.294,-2.269,-5.054,-6.81,-6.035,-4.465,-6.113,-6.537,-3.416,-2.261,-5.855,-5.193,-1.649,-5.12,-2.342,-4.617,-5.331,-6.92,-2.36,-3.589,-7.287,-3.878,-4.711,-5.117,-4.505,-5.835,-7.19,-4.085,-5.859,-5.576,-6.848,-4.881,-3.961,-2.24,-5.296,-6.853,-6.002,-6.846,-6.879,-6.404,-6.851,-7.163,-6.631,-3.082,-7.452,-1.704,-7.0,-2.196,-4.842,-5.352,-2.772,-6.514,-5.483,-4.052,-0.656,-5.49,-4.172,-4.032,-4.402,-2.645,-3.587,-3.967,-6.08,-3.788,-7.087,-7.091,-3.307,-3.905,-4.278,-5.457,-6.417,-7.354,-5.021,-7.147,-3.309,-6.034,-6.501,-4.932,-3.774,-4.067,-4.735,-5.743,-1.885,-5.563,-3.428,-6.225,-4.056,-7.377,-3.661,-4.368,-3.474,-5.738,-5.712,-4.933,-3.029,-3.438,-6.656,-5.117,-2.039,-5.541,-4.755,-6.711,-7.294,-4.084,-6.396,-0.409,-4.286,-2.781,-5.03,-5.652,-6.609,-6.977,-7.094,-3.833,-0.032,-6.704,-6.847,-1.923,-5.294,-7.273,-6.088,-4.102,-3.975,-4.164,-3.318,-4.576,-2.676,-6.022,-5.601,-5.643,-5.525,-3.023,-6.079,-4.865,-3.99,-4.722,-5.823,-3.726,-5.238,-3.963,-2.898,-3.349,-5.426,-5.61,-5.778,-4.291,-7.498,-0.102,-4.85,-6.436,-2.619,-3.935,-5.406,-4.897,-3.092,-2.964,-5.346,-3.307,-5.7,-3.231,-6.14,-6.713,-5.691,-7.045,-5.403,-4.323,-5.201,-6.813,-3.955,-2.163,-2.724,-4.913,-6.316,-7.303,-6.888,-6.332],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"below threshold=yes\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"yes","marker":{"color":"#EF553B","opacity":1,"pattern":{"shape":""}},"name":"yes","nbinsx":20,"offsetgroup":"yes","orientation":"v","showlegend":true,"x":[-9.24,-8.665,-7.574,-7.917,-8.163,-13.305,-10.512,-7.916,-8.534,-7.886,-9.789,-15.265,-10.476,-11.891,-8.499,-14.373,-8.062,-8.676,-9.04,-9.708,-10.182,-9.738,-8.31,-11.722,-15.247,-13.009,-10.524,-11.179,-9.417,-9.76,-9.624,-8.668,-8.922,-10.988,-7.562,-11.494,-9.311,-11.546,-7.82,-10.55,-9.838,-9.247,-10.043,-8.456,-12.181,-12.991,-12.021,-9.904,-11.098,-13.547,-9.939,-10.267,-12.241,-14.01,-8.335,-9.771,-8.273,-9.418,-11.672,-9.5,-8.853,-10.275,-9.819,-9.548,-11.077,-11.328,-9.742,-13.477,-10.657,-8.58,-8.015,-13.313,-11.903,-11.24,-12.293,-11.623,-12.76,-13.53,-13.247,-13.102,-12.073,-14.094,-8.465,-11.546,-11.198,-8.086,-11.091,-9.862,-13.533,-8.62,-11.427,-7.574,-11.599,-10.674,-10.946,-13.568,-8.637,-8.05,-11.013,-14.583,-12.225,-14.027,-12.347,-10.452,-7.88,-12.914,-8.472,-8.889,-11.494,-10.518,-9.659,-13.291,-11.674,-9.316,-8.399,-8.697,-8.541,-9.049,-13.129,-9.681,-11.629,-10.342,-11.822,-10.11,-12.838,-13.534,-7.662,-10.334,-13.632,-12.706,-12.976,-7.879,-11.443,-8.365,-9.676,-12.839,-11.058,-9.07,-8.027,-9.008,-8.0,-13.131,-9.68,-15.569,-11.647,-8.785,-8.965,-7.556,-11.751,-8.08,-9.641,-12.069,-9.275,-10.949,-10.87,-11.428,-12.238,-12.902,-10.938,-11.083,-11.256,-7.578,-12.091,-8.489,-9.929,-8.817,-15.482,-11.748,-11.981,-10.585,-10.848,-11.517,-11.661,-9.824,-11.947,-10.392,-9.367,-11.271,-8.422,-11.223,-8.817,-8.737,-10.995,-11.932,-10.094,-11.256,-11.52,-9.854,-10.727,-13.123,-11.193,-10.97,-13.024,-12.188,-14.2,-8.416,-8.411,-7.704,-11.374,-12.9,-13.95,-12.562,-9.453,-8.363,-7.852,-10.377,-7.991,-10.563,-10.447,-8.282,-9.9,-7.508,-7.762,-8.468,-11.303,-9.586,-12.56,-9.487,-13.105,-10.482,-11.55,-9.091,-8.725,-9.726,-12.158,-9.024,-10.663,-10.15,-10.273,-9.522,-8.242,-7.635,-9.373,-9.153,-7.676,-13.393,-9.851,-12.62,-12.029,-8.022,-12.013,-8.943,-7.658,-11.552,-14.427,-9.083,-11.157,-8.77,-10.136,-8.306,-8.273,-7.741,-8.565,-11.868,-8.426,-7.655,-9.3,-8.171,-11.548,-8.497,-7.752,-10.428,-11.863,-7.893,-8.689,-8.848,-7.854,-7.726,-9.012,-8.291,-8.101,-13.475,-11.549,-8.22,-8.335,-9.116,-8.627,-11.381,-12.17,-12.786,-8.47,-10.559,-11.542,-12.716,-11.963,-14.742,-10.615,-8.785,-11.04,-7.967,-11.117,-14.417,-13.017,-9.716,-8.779,-8.263,-8.536,-9.33,-10.854,-8.052,-9.6,-7.557,-8.242,-9.889,-11.968,-9.061,-12.852,-12.468,-7.641,-8.399,-10.492,-9.247,-8.812,-13.295,-9.022,-9.792,-12.924,-11.328,-10.777,-9.018,-12.728,-12.849,-8.444,-9.314,-11.753,-10.268,-10.353,-11.109,-11.36,-9.486,-9.201,-9.036,-10.871,-11.589,-15.203,-12.431,-11.373,-10.986,-9.601,-9.692,-10.412,-10.756,-13.468,-8.088,-16.026,-11.124,-12.054,-10.46,-13.77,-10.232,-10.313,-9.186,-9.476,-11.638,-12.864,-11.669,-12.337,-7.901,-10.341,-13.021,-9.337,-12.035,-8.054,-9.233,-9.602,-14.151,-8.801,-14.893,-10.118,-11.75,-12.047,-11.265,-8.357,-11.6,-9.324,-12.087,-9.151,-9.162,-11.664,-11.478,-7.558,-9.954,-8.792,-11.965,-7.806,-11.3,-10.734,-10.615,-11.081,-8.804,-11.892,-10.28,-12.073,-7.619,-12.423,-13.225,-11.773,-9.801,-13.359,-12.057,-12.298,-7.669,-10.797,-9.935,-7.86,-8.258,-10.632,-7.874,-10.559,-10.22,-13.695,-14.58,-7.88,-12.974,-9.019,-7.772,-8.035,-11.006,-13.892,-8.158,-10.313,-7.88,-10.531,-13.687,-11.658,-10.563,-11.504,-8.949,-11.694,-11.627,-9.643,-10.4,-9.002,-8.994,-8.427,-11.391,-13.092,-11.536,-8.956,-8.007,-11.63,-10.165,-7.669,-8.013,-8.974,-10.559,-9.658,-9.173,-8.037,-10.074,-11.076,-7.602,-13.272,-12.432,-10.839,-10.279,-12.282,-9.286,-7.954,-13.656,-10.33,-12.237,-10.782,-14.762,-9.651,-8.489,-9.816,-9.398,-7.686,-11.034,-9.913,-12.377,-7.766,-7.89,-9.394,-11.12,-8.22,-11.2,-10.235,-9.426,-8.413,-11.158,-11.265,-10.881,-12.403,-9.493,-8.772,-10.519,-9.713,-8.218,-9.031,-14.119,-12.776,-7.759,-12.196,-11.898,-13.732,-11.867,-9.465,-8.643,-13.06,-10.139,-7.957,-7.779,-7.833,-11.409,-10.97,-8.151,-7.811,-9.097,-8.876,-11.577,-10.876,-7.663,-10.808,-10.188,-9.538,-11.336,-11.228,-8.045,-10.217,-7.857,-9.734,-9.506,-9.528,-10.196,-12.899,-10.206,-8.397,-16.797,-8.858],"xaxis":"x","yaxis":"y","type":"histogram"}],                        {"template":{"data":{"histogram2dcontour":[{"type":"histogram2dcontour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"choropleth":[{"type":"choropleth","colorbar":{"outlinewidth":0,"ticks":""}}],"histogram2d":[{"type":"histogram2d","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmap":[{"type":"heatmap","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmapgl":[{"type":"heatmapgl","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"contourcarpet":[{"type":"contourcarpet","colorbar":{"outlinewidth":0,"ticks":""}}],"contour":[{"type":"contour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"surface":[{"type":"surface","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"mesh3d":[{"type":"mesh3d","colorbar":{"outlinewidth":0,"ticks":""}}],"scatter":[{"fillpattern":{"fillmode":"overlay","size":10,"solidity":0.2},"type":"scatter"}],"parcoords":[{"type":"parcoords","line":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolargl":[{"type":"scatterpolargl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"bar":[{"error_x":{"color":"#2a3f5f"},"error_y":{"color":"#2a3f5f"},"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"bar"}],"scattergeo":[{"type":"scattergeo","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolar":[{"type":"scatterpolar","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"histogram":[{"marker":{"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"histogram"}],"scattergl":[{"type":"scattergl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatter3d":[{"type":"scatter3d","line":{"colorbar":{"outlinewidth":0,"ticks":""}},"marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattermapbox":[{"type":"scattermapbox","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterternary":[{"type":"scatterternary","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattercarpet":[{"type":"scattercarpet","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"carpet":[{"aaxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"baxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"type":"carpet"}],"table":[{"cells":{"fill":{"color":"#EBF0F8"},"line":{"color":"white"}},"header":{"fill":{"color":"#C8D4E3"},"line":{"color":"white"}},"type":"table"}],"barpolar":[{"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"barpolar"}],"pie":[{"automargin":true,"type":"pie"}]},"layout":{"autotypenumbers":"strict","colorway":["#636efa","#EF553B","#00cc96","#ab63fa","#FFA15A","#19d3f3","#FF6692","#B6E880","#FF97FF","#FECB52"],"font":{"color":"#2a3f5f"},"hovermode":"closest","hoverlabel":{"align":"left"},"paper_bgcolor":"white","plot_bgcolor":"#E5ECF6","polar":{"bgcolor":"#E5ECF6","angularaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"radialaxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"ternary":{"bgcolor":"#E5ECF6","aaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"baxis":{"gridcolor":"white","linecolor":"white","ticks":""},"caxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"coloraxis":{"colorbar":{"outlinewidth":0,"ticks":""}},"colorscale":{"sequential":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"sequentialminus":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"diverging":[[0,"#8e0152"],[0.1,"#c51b7d"],[0.2,"#de77ae"],[0.3,"#f1b6da"],[0.4,"#fde0ef"],[0.5,"#f7f7f7"],[0.6,"#e6f5d0"],[0.7,"#b8e186"],[0.8,"#7fbc41"],[0.9,"#4d9221"],[1,"#276419"]]},"xaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"yaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"scene":{"xaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"yaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"zaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2}},"shapedefaults":{"line":{"color":"#2a3f5f"}},"annotationdefaults":{"arrowcolor":"#2a3f5f","arrowhead":0,"arrowwidth":1},"geo":{"bgcolor":"white","landcolor":"#E5ECF6","subunitcolor":"white","showland":true,"showlakes":true,"lakecolor":"white"},"title":{"x":0.05},"mapbox":{"style":"light"}}},"xaxis":{"anchor":"y","domain":[0.0,1.0],"title":{"text":"Scores"}},"yaxis":{"anchor":"x","domain":[0.0,1.0],"title":{"text":"count"}},"legend":{"title":{"text":"below threshold"},"tracegroupgap":0},"title":{"text":"Distribution of Scores Within Gnomad"},"barmode":"relative","shapes":[{"line":{"color":"red"},"opacity":0.4,"type":"line","x0":-11.661,"x1":-11.661,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"red"},"opacity":0.4,"type":"line","x0":-10.267,"x1":-10.267,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"red"},"opacity":0.4,"type":"line","x0":-8.792,"x1":-8.792,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-11.627,"x1":-11.627,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-10.531,"x1":-10.531,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-9.097,"x1":-9.097,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-8.77,"x1":-8.77,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-8.397,"x1":-8.397,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-7.857,"x1":-7.857,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-7.578,"x1":-7.578,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-7.14,"x1":-7.14,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.877,"x1":-6.877,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.744,"x1":-6.744,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.691,"x1":-6.691,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.417,"x1":-6.417,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.373,"x1":-6.373,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-6.256,"x1":-6.256,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-5.5,"x1":-5.5,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-5.457,"x1":-5.457,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-5.382,"x1":-5.382,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-5.089,"x1":-5.089,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-4.799,"x1":-4.799,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-4.738,"x1":-4.738,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-4.634,"x1":-4.634,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-3.96,"x1":-3.96,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-3.652,"x1":-3.652,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-2.664,"x1":-2.664,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-2.647,"x1":-2.647,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-2.449,"x1":-2.449,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-1.885,"x1":-1.885,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-1.597,"x1":-1.597,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-1.22,"x1":-1.22,"xref":"x","y0":0,"y1":1,"yref":"paper"},{"line":{"color":"green"},"opacity":0.4,"type":"line","x0":-0.146,"x1":-0.146,"xref":"x","y0":0,"y1":1,"yref":"paper"}],"bargap":0.1},                        {"responsive": true}                    ).then(function(){
                            
var gd = document.getElementById('49b4255d-0485-43ad-bc98-bc33bb1320da');
var x = new MutationObserver(function (mutations, observer) {{
        var display = window.getComputedStyle(gd).display;
        if (!display || display === 'none') {{
            console.log([gd, 'removed!']);
            Plotly.purge(gd);
            observer.disconnect();
        }}
}});

// Listen for the removal of the full notebook cells
var notebookContainer = gd.closest('#notebook-container');
if (notebookContainer) {{
    x.observe(notebookContainer, {childList: true});
}}

// Listen for the clearing of the current output cell
var outputEl = gd.closest('.output');
if (outputEl) {{
    x.observe(outputEl, {childList: true});
}}

                        })                };                });            </script>        </div>
</div>
</div>
<p>This is the most informative figure in this preliminary analysis. We see that, yes, the benign tend to be greater than -7.5; but there are more benign samples below the -7.5 threshold than there are pathogenic samples, and overall we have very limited data here to work with. The threshold can’t be proven in our circumstance to be useful. There is at least a trend for benign variants to be above the threshold.</p>
<p>But we do have recorded variants at the lowest score that we could start examining first, and at least identify some new variants that can be added. These can be modeled in AlphaFold and then examined manually, and put through the simulation pipeline. <!-- 
I'll also include a graph here -- this is all gnomAD variants, plotted with ClinVar Clinical Significance. The colors I could play with... and you can toggle data. Interesting to visualize just how many uncertain significance stuff there is. But the fact of the VERY small pathogenic group can only serve to bias, I don't think this graph is scientifically healthy to look at.

::: {.cell execution_count=19}
``` {.python .cell-code}
fig = px.histogram(
    merged_df,
    x="score",
    color = "ClinVar Clinical Significance",
    nbins=20,
    opacity=1,
    labels={"score": "Scores"},
    title="Distribution of Scores Within Gnomad",
)

# # Customize the layout
fig.update_layout(bargap=0.1)

# # Show the plot
fig.show()
```

::: {.cell-output .cell-output-display}

```{=html}
<div>                            <div id="ae53ba39-6657-4d39-a497-c2699ea31815" class="plotly-graph-div" style="height:525px; width:100%;"></div>            <script type="text/javascript">                require(["plotly"], function(Plotly) {                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById("ae53ba39-6657-4d39-a497-c2699ea31815")) {                    Plotly.newPlot(                        "ae53ba39-6657-4d39-a497-c2699ea31815",                        [{"alignmentgroup":"True","bingroup":"x","hovertemplate":"ClinVar Clinical Significance=Likely benign\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"Likely benign","marker":{"color":"#EF553B","opacity":1,"pattern":{"shape":""}},"name":"Likely benign","nbinsx":20,"offsetgroup":"Likely benign","orientation":"v","showlegend":true,"x":[-2.664,-5.089,-3.652,-4.634,-6.256,-1.22,-2.449,-7.14,-7.578,-6.744,-6.373,-8.77,-1.597,-6.691,-5.382,-5.5,-10.531,-11.627,-7.857],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"ClinVar Clinical Significance=Benign\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"Benign","marker":{"color":"#00cc96","opacity":1,"pattern":{"shape":""}},"name":"Benign","nbinsx":20,"offsetgroup":"Benign","orientation":"v","showlegend":true,"x":[-2.647,-4.799,-0.146,-3.96,-6.877,-6.417,-8.397],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"ClinVar Clinical Significance=Conflicting interpretations of pathogenicity\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"Conflicting interpretations of pathogenicity","marker":{"color":"#ab63fa","opacity":1,"pattern":{"shape":""}},"name":"Conflicting interpretations of pathogenicity","nbinsx":20,"offsetgroup":"Conflicting interpretations of pathogenicity","orientation":"v","showlegend":true,"x":[-5.614,-5.018,-10.512,-2.504,-9.771,-11.443,-6.629,-8.363,-10.663,-3.015,-8.426,-10.428,-4.553,-2.651,-3.026,-10.854,-12.035,-4.175,-6.848,-11.63,-4.368,-1.923,-9.506],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"ClinVar Clinical Significance=Uncertain significance\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"Uncertain significance","marker":{"color":"#FFA15A","opacity":1,"pattern":{"shape":""}},"name":"Uncertain significance","nbinsx":20,"offsetgroup":"Uncertain significance","orientation":"v","showlegend":true,"x":[-2.097,-2.583,-1.665,-7.917,-8.163,-3.553,-5.233,-7.886,-5.32,-6.241,-11.722,-10.524,-9.624,-8.922,-10.988,-7.562,-5.901,-3.792,-4.465,-2.136,-5.563,-6.252,-8.273,-5.819,-13.53,-11.091,-7.574,-7.104,-13.568,-8.637,-12.225,-12.347,-6.103,-8.472,-5.597,-6.458,-4.164,-6.279,-3.31,-1.678,-10.342,-3.703,-4.311,-4.765,-10.334,-6.974,-6.777,-6.039,-8.365,-4.421,-12.839,-9.07,-5.507,-13.131,-4.496,-8.965,-7.046,-6.813,-6.675,-2.514,-3.512,-7.034,-8.817,-5.665,-4.654,-6.166,-11.517,-2.104,-5.322,-11.223,-8.817,0.526,-10.094,-7.498,-9.854,-5.663,-11.193,-7.704,-6.421,-5.755,-10.563,-8.282,-6.684,-5.873,-3.356,-11.55,-3.724,-8.725,-9.726,-12.158,-10.273,-7.145,-6.563,-9.851,-6.611,-5.39,-9.083,-11.157,-3.973,-3.563,-7.17,-7.154,-6.937,-2.751,-5.54,-8.171,-8.497,-2.21,-1.801,-3.403,-4.442,-7.054,-6.322,-6.279,-6.652,-4.539,-5.998,-8.848,-7.854,-4.931,-5.401,-5.362,-8.291,-13.475,-11.549,-8.335,-9.116,-12.17,-6.425,-8.785,-7.967,-11.117,-7.362,-1.816,-8.052,-9.6,-6.409,-5.765,-6.92,-5.404,-13.295,-11.328,-5.937,-12.849,-5.747,-11.589,-12.431,-3.646,-9.692,-10.412,-12.054,-9.476,-7.375,-12.864,-12.337,-10.341,-7.109,-3.795,-5.17,-6.923,-8.357,-6.932,-9.954,-10.615,-8.804,-4.068,-11.892,-7.619,-4.868,-7.669,-10.797,-7.467,-5.213,-7.874,-10.559,-14.58,-3.416,-7.772,-2.342,-11.006,-3.589,-7.287,-3.878,-9.643,-8.994,-3.961,-8.956,-10.074,-13.272,-10.839,-2.196,-5.352,-9.286,-10.33,-12.237,-9.651,-7.686,-9.913,-9.394,-3.788,-11.2,-9.426,-8.413,-11.158,-3.307,-10.519,-8.218,-12.196,-6.501,-3.774,-4.067,-10.139,-5.563,-4.056,-7.377,-3.661,-5.712,-4.933,-4.084,-6.396,-8.045,-0.032,-5.294,-4.164,-9.734,-5.238,-2.898,-5.778,-4.291,-7.498,-4.897,-6.713,-12.899,-10.206,-4.913,-6.332],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"ClinVar Clinical Significance=Benign\u002fLikely benign\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"Benign\u002fLikely benign","marker":{"color":"#19d3f3","opacity":1,"pattern":{"shape":""}},"name":"Benign\u002fLikely benign","nbinsx":20,"offsetgroup":"Benign\u002fLikely benign","orientation":"v","showlegend":true,"x":[-4.738,-5.457,-1.885,-9.097],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"ClinVar Clinical Significance=Likely pathogenic\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"Likely pathogenic","marker":{"color":"#FF6692","opacity":1,"pattern":{"shape":""}},"name":"Likely pathogenic","nbinsx":20,"offsetgroup":"Likely pathogenic","orientation":"v","showlegend":true,"x":[-10.267,-8.792],"xaxis":"x","yaxis":"y","type":"histogram"},{"alignmentgroup":"True","bingroup":"x","hovertemplate":"ClinVar Clinical Significance=Pathogenic\u003cbr\u003eScores=%{x}\u003cbr\u003ecount=%{y}\u003cextra\u003e\u003c\u002fextra\u003e","legendgroup":"Pathogenic","marker":{"color":"#B6E880","opacity":1,"pattern":{"shape":""}},"name":"Pathogenic","nbinsx":20,"offsetgroup":"Pathogenic","orientation":"v","showlegend":true,"x":[-11.661],"xaxis":"x","yaxis":"y","type":"histogram"}],                        {"template":{"data":{"histogram2dcontour":[{"type":"histogram2dcontour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"choropleth":[{"type":"choropleth","colorbar":{"outlinewidth":0,"ticks":""}}],"histogram2d":[{"type":"histogram2d","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmap":[{"type":"heatmap","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"heatmapgl":[{"type":"heatmapgl","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"contourcarpet":[{"type":"contourcarpet","colorbar":{"outlinewidth":0,"ticks":""}}],"contour":[{"type":"contour","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"surface":[{"type":"surface","colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]}],"mesh3d":[{"type":"mesh3d","colorbar":{"outlinewidth":0,"ticks":""}}],"scatter":[{"fillpattern":{"fillmode":"overlay","size":10,"solidity":0.2},"type":"scatter"}],"parcoords":[{"type":"parcoords","line":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolargl":[{"type":"scatterpolargl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"bar":[{"error_x":{"color":"#2a3f5f"},"error_y":{"color":"#2a3f5f"},"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"bar"}],"scattergeo":[{"type":"scattergeo","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterpolar":[{"type":"scatterpolar","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"histogram":[{"marker":{"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"histogram"}],"scattergl":[{"type":"scattergl","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatter3d":[{"type":"scatter3d","line":{"colorbar":{"outlinewidth":0,"ticks":""}},"marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattermapbox":[{"type":"scattermapbox","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scatterternary":[{"type":"scatterternary","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"scattercarpet":[{"type":"scattercarpet","marker":{"colorbar":{"outlinewidth":0,"ticks":""}}}],"carpet":[{"aaxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"baxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"type":"carpet"}],"table":[{"cells":{"fill":{"color":"#EBF0F8"},"line":{"color":"white"}},"header":{"fill":{"color":"#C8D4E3"},"line":{"color":"white"}},"type":"table"}],"barpolar":[{"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"barpolar"}],"pie":[{"automargin":true,"type":"pie"}]},"layout":{"autotypenumbers":"strict","colorway":["#636efa","#EF553B","#00cc96","#ab63fa","#FFA15A","#19d3f3","#FF6692","#B6E880","#FF97FF","#FECB52"],"font":{"color":"#2a3f5f"},"hovermode":"closest","hoverlabel":{"align":"left"},"paper_bgcolor":"white","plot_bgcolor":"#E5ECF6","polar":{"bgcolor":"#E5ECF6","angularaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"radialaxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"ternary":{"bgcolor":"#E5ECF6","aaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"baxis":{"gridcolor":"white","linecolor":"white","ticks":""},"caxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"coloraxis":{"colorbar":{"outlinewidth":0,"ticks":""}},"colorscale":{"sequential":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"sequentialminus":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"diverging":[[0,"#8e0152"],[0.1,"#c51b7d"],[0.2,"#de77ae"],[0.3,"#f1b6da"],[0.4,"#fde0ef"],[0.5,"#f7f7f7"],[0.6,"#e6f5d0"],[0.7,"#b8e186"],[0.8,"#7fbc41"],[0.9,"#4d9221"],[1,"#276419"]]},"xaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"yaxis":{"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","automargin":true,"zerolinewidth":2},"scene":{"xaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"yaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2},"zaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white","gridwidth":2}},"shapedefaults":{"line":{"color":"#2a3f5f"}},"annotationdefaults":{"arrowcolor":"#2a3f5f","arrowhead":0,"arrowwidth":1},"geo":{"bgcolor":"white","landcolor":"#E5ECF6","subunitcolor":"white","showland":true,"showlakes":true,"lakecolor":"white"},"title":{"x":0.05},"mapbox":{"style":"light"}}},"xaxis":{"anchor":"y","domain":[0.0,1.0],"title":{"text":"Scores"}},"yaxis":{"anchor":"x","domain":[0.0,1.0],"title":{"text":"count"}},"legend":{"title":{"text":"ClinVar Clinical Significance"},"tracegroupgap":0},"title":{"text":"Distribution of Scores Within Gnomad"},"barmode":"relative","bargap":0.1},                        {"responsive": true}                    ).then(function(){
                            
var gd = document.getElementById('ae53ba39-6657-4d39-a497-c2699ea31815');
var x = new MutationObserver(function (mutations, observer) {{
        var display = window.getComputedStyle(gd).display;
        if (!display || display === 'none') {{
            console.log([gd, 'removed!']);
            Plotly.purge(gd);
            observer.disconnect();
        }}
}});

// Listen for the removal of the full notebook cells
var notebookContainer = gd.closest('#notebook-container');
if (notebookContainer) {{
    x.observe(notebookContainer, {childList: true});
}}

// Listen for the clearing of the current output cell
var outputEl = gd.closest('.output');
if (outputEl) {{
    x.observe(outputEl, {childList: true});
}}

                        })                };                });            </script>        </div>
```

:::
:::


---></p>
<p>And so, for the sake of seeing what this most deleterious variant can demonstrate if modeled in AlphaFold… let’s get the lowest scored missense mutation is.</p>
<div class="cell" data-execution_count="20">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Lowest score:"</span>, merged_df.score.<span class="bu">min</span>())</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>merged_df[[<span class="st">'variant'</span>,<span class="st">'score'</span>]].tail()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Lowest score: -16.797</code></pre>
</div>
<div class="cell-output cell-output-display" data-execution_count="20">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">variant</th>
<th data-quarto-table-cell-role="th">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">1136</td>
<td>L13P</td>
<td>-16.797</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1137</td>
<td>S12P</td>
<td>-8.858</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1138</td>
<td>F6C</td>
<td>-7.303</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1139</td>
<td>R3I</td>
<td>-6.888</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1140</td>
<td>A2E</td>
<td>-6.332</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>If we look for this mutation in P14410 in UniProt, https://www.ebi.ac.uk/ProtVar/query?search=P14410%20L13P it’s also predicted deleterious by CADD, a model from 2021 https://cadd.gs.washington.edu/info. We can also cross reference with this model in future, but ESM1b in its paper demonstrates superiority.</p>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body" data-line-spacing="2" role="list">
<div id="ref-weijersDiarrhoeaCausedDeficiency1960" class="csl-entry" role="listitem">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Weijers, H. A., va de KAMER, J. H., Mossel, D. A. &amp; Dicke, W. K. <a href="https://doi.org/10.1016/s0140-6736(60)91381-7">Diarrhoea caused by deficiency of sugar-splitting enzymes</a>. <em>Lancet (London, England)</em> <strong>2</strong>, 296–297 (1960).</div>
</div>
<div id="ref-welshIntestinalDisaccharidaseActivities1978" class="csl-entry" role="listitem">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Welsh, J. D., Poley, J. R., Bhatia, M. &amp; Stevenson, D. E. <a href="https://www.ncbi.nlm.nih.gov/pubmed/100368">Intestinal disaccharidase activities in relation to age, race, and mucosal damage</a>. <em>Gastroenterology</em> <strong>75</strong>, 847–855 (1978).</div>
</div>
<div id="ref-ellestad-sayedDisaccharideMalabsorptionDietary1978" class="csl-entry" role="listitem">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Ellestad-Sayed, J., Haworth, J. &amp; Hildes, J. <a href="https://doi.org/10.1093/ajcn/31.8.1473">Disaccharide malabsorption and dietary patterns in two <span>Canadian Eskimo</span> Communities1234</a>. <em>The American Journal of Clinical Nutrition</em> <strong>31</strong>, 1473–1478 (1978).</div>
</div>
<div id="ref-gudmand-hoyerSucraseDeficiencyGreenland1987" class="csl-entry" role="listitem">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Gudmand-Høyer, E., Fenger, H. J., Kern-Hansen, P. &amp; Madsen, P. R. <a href="https://doi.org/10.3109/00365528708991851">Sucrase deficiency in <span>Greenland</span>. <span>Incidence</span> and genetic aspects</a>. <em>Scandinavian Journal of Gastroenterology</em> <strong>22</strong>, 24–28 (1987).</div>
</div>
<div id="ref-henstromFunctionalVariantsSucrase2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Henström, M. <em>et al.</em> <a href="https://doi.org/10.1136/gutjnl-2016-312456">Functional variants in the sucraseisomaltase gene associate with increased risk of irritable bowel syndrome</a>. <em>Gut</em> <strong>67</strong>, 263–270 (2018).</div>
</div>
<div id="ref-garcia-etxebarriaIncreasedPrevalenceRare2018a" class="csl-entry" role="listitem">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Garcia-Etxebarria, K. <em>et al.</em> <a href="https://doi.org/10.1016/j.cgh.2018.01.047">Increased <span>Prevalence</span> of <span class="nocase">Rare Sucrase-isomaltase Pathogenic Variants</span> in <span>Irritable Bowel Syndrome Patients</span></a>. <em>Clinical Gastroenterology and Hepatology</em> <strong>16</strong>, 1673–1676 (2018).</div>
</div>
<div id="ref-kimSucraseIsomaltaseDeficiencyPotential2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Kim, S. B., Calmet, F. H., Garrido, J., Garcia-Buitrago, M. T. &amp; Moshiree, B. <a href="https://doi.org/10.1007/s10620-019-05780-7">Sucrase-<span>Isomaltase Deficiency</span> as a <span>Potential Masquerader</span> in <span>Irritable Bowel Syndrome</span></a>. <em>Digestive Diseases and Sciences</em> <strong>65</strong>, 534–540 (2020).</div>
</div>
<div id="ref-huseinHeterozygotesArePotential2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Husein, D. M., Wanes, D., Marten, L. M., Zimmer, K.-P. &amp; Naim, H. Y. <a href="https://doi.org/10.3390/nu11102290">Heterozygotes <span>Are</span> a <span>Potential New Entity</span> among <span>Homozygotes</span> and <span>Compound Heterozygotes</span> in <span>Congenital Sucrase-Isomaltase Deficiency</span></a>. <em>Nutrients</em> <strong>11</strong>, 2290 (2019).</div>
</div>
<div id="ref-habermanCongenitalSucraseisomaltaseDeficiency2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Haberman, Y. <em>et al.</em> <a href="https://doi.org/10.1097/MPG.0000000000001424">Congenital <span class="nocase">Sucrase-isomaltase Deficiency</span>: <span>A Novel Compound Heterozygous Mutation Causing Aberrant Protein Localization</span></a>. <em>Journal of pediatric gastroenterology and nutrition</em> <strong>64</strong>, 770–776 (2017).</div>
</div>
<div id="ref-taskinPrevalenceCongenitalSucraseisomaltase2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Taskin, D. G. <em>et al.</em> <a href="https://doi.org/10.1007/s12041-023-01428-8">Prevalence of congenital sucrase-isomaltase deficiency in <span>Turkey</span> may be much higher than the estimates</a>. <em>Journal of Genetics</em> <strong>102</strong>, 31 (2023).</div>
</div>
<div id="ref-chenGenomewideMutationalConstraint2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Chen, S. <em>et al.</em> A genome-wide mutational constraint map quantified from variation in 76,156 human genomes. 2022.03.20.485034 (2022) doi:<a href="https://doi.org/10.1101/2022.03.20.485034">10.1101/2022.03.20.485034</a>.</div>
</div>
<div id="ref-landrumClinVarPublicArchive2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Landrum, M. J. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkv1222"><span>ClinVar</span>: Public archive of interpretations of clinically relevant variants</a>. <em>Nucleic Acids Research</em> <strong>44</strong>, D862–D868 (2016).</div>
</div>
<div id="ref-deleusseEstimatingPrevalenceCongenital2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">De Leusse, C., Roman, C., Roquelaure, B. &amp; Fabre, A. <a href="https://doi.org/10.1016/j.arcped.2022.08.005">Estimating the prevalence of congenital disaccharidase deficiencies using allele frequencies from <span class="nocase">gnomAD</span></a>. <em>Archives de Pédiatrie</em> <strong>29</strong>, 599–603 (2022).</div>
</div>
<div id="ref-mclarenEnsemblVariantEffect2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">McLaren, W. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-016-0974-4">The <span>Ensembl Variant Effect Predictor</span></a>. <em>Genome Biology</em> <strong>17</strong>, 122 (2016).</div>
</div>
<div id="ref-brandesGenomewidePredictionDisease2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Brandes, N., Goldman, G., Wang, C. H., Ye, C. J. &amp; Ntranos, V. Genome-wide prediction of disease variant effects with a deep protein language model. <em>Nature Genetics</em> (2023) doi:<a href="https://doi.org/10.1038/s41588-023-01465-0">10.1038/s41588-023-01465-0</a>.</div>
</div>
<div id="ref-kurosawaPDIVASPathogenicityPredictor2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Kurosawa, R. <em>et al.</em> <span>PDIVAS</span>: <span>Pathogenicity</span> predictor for <span>Deep-Intronic Variants</span> causing <span>Aberrant Splicing</span>. 2023.03.20.23287464 (2023) doi:<a href="https://doi.org/10.1101/2023.03.20.23287464">10.1101/2023.03.20.23287464</a>.</div>
</div>
<div id="ref-jumperHighlyAccurateProtein2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Jumper, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-021-03819-2">Highly accurate protein structure prediction with <span>AlphaFold</span></a>. <em>Nature</em> <strong>596</strong>, 583–589 (2021).</div>
</div>
<div id="ref-corsoDiffDockDiffusionSteps2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Corso, G., Stärk, H., Jing, B., Barzilay, R. &amp; Jaakkola, T. <span>DiffDock</span>: <span>Diffusion Steps</span>, <span>Twists</span>, and <span>Turns</span> for <span>Molecular Docking</span>. (2023) doi:<a href="https://doi.org/10.48550/arXiv.2210.01776">10.48550/arXiv.2210.01776</a>.</div>
</div>
<div id="ref-ketataDiffDockPPRigidProteinProtein2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Ketata, M. A. <em>et al.</em> <a href="https://arxiv.org/abs/2304.03889"><span>DiffDock-PP</span>: <span>Rigid Protein-Protein Docking</span> with <span>Diffusion Models</span></a>. (2023).</div>
</div>
<div id="ref-brocidiaconoPLANTAINDiffusioninspiredPose2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Brocidiacono, M., Popov, K. I., Koes, D. R. &amp; Tropsha, A. <a href="https://arxiv.org/abs/2307.12090"><span>PLANTAIN</span>: <span class="nocase">Diffusion-inspired Pose Score Minimization</span> for <span>Fast</span> and <span>Accurate Molecular Docking</span></a>. (2023).</div>
</div>
<div id="ref-trottAutoDockVinaImproving2010" class="csl-entry" role="listitem">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Trott, O. &amp; Olson, A. J. <a href="https://doi.org/10.1002/jcc.21334"><span>AutoDock Vina</span>: Improving the speed and accuracy of docking with a new scoring function, efficient optimization and multithreading</a>. <em>Journal of computational chemistry</em> <strong>31</strong>, 455–461 (2010).</div>
</div>
<div id="ref-santos-martinsAcceleratingAutoDock4GPUs2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Santos-Martins, D. <em>et al.</em> <a href="https://doi.org/10.1021/acs.jctc.0c01006">Accelerating <span>AutoDock4</span> with <span>GPUs</span> and <span>Gradient-Based Local Search</span></a>. <em>Journal of Chemical Theory and Computation</em> <strong>17</strong>, 1060–1073 (2021).</div>
</div>
<div id="ref-robertsFoldyWebApplication2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Roberts, J. B. <em>et al.</em> Foldy: A web application for interactive protein structure analysis. (2023) doi:<a href="https://doi.org/10.1101/2023.05.11.540333">10.1101/2023.05.11.540333</a>.</div>
</div>
<div id="ref-wongBenchmarkingAlphaFoldenabledMolecular2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Wong, F. <em>et al.</em> <a href="https://doi.org/10.15252/msb.202211081">Benchmarking <span class="nocase">AlphaFold-enabled</span> molecular docking predictions for antibiotic discovery</a>. <em>Molecular Systems Biology</em> <strong>18</strong>, e11081 (2022).</div>
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>